{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "aa0f0d95-5e6b-436f-a971-9486d0841329",
   "metadata": {},
   "source": [
    "# Data collection and project setup\n",
    "\n",
    "Summary of data collection steps followed in the Streptomyces pangenome project. This notebook describes all the projects and BGCFlow runs carried out in the study.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d21448bb-afe9-4fdb-b964-2143a0ccd1e4",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from pathlib import Path\n",
    "import altair as alt\n",
    "import yaml\n",
    "import warnings\n",
    "import matplotlib.pyplot as plt\n",
    "warnings.filterwarnings('ignore')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a46880a2-ae00-465d-a4c0-5b2fc4c41391",
   "metadata": {},
   "source": [
    "### File Configuration"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "bab09367-1019-4c03-9401-9693de605473",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'bgcflow_dir': '/datadrive/bgcflow'}"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "with open(\"config.yaml\", \"r\") as f:\n",
    "    notebook_configuration = yaml.safe_load(f)\n",
    "notebook_configuration"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8a3745d8-3c30-4792-aabd-5a1b10a0370b",
   "metadata": {},
   "source": [
    "# 1. Setup qc project with all genomes from NCBI\n",
    "\n",
    "2938 genome accessions were collected from the NCBI RefSeq database on the 30-06-2023 for genomes belonging to Streptomycetaceae family"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "cb22bdea-1486-45e3-a1c3-499c0b6e20d1",
   "metadata": {},
   "outputs": [],
   "source": [
    "tables_dir = Path(\"assets/tables/\")\n",
    "ncbi_accn_table = tables_dir / \"streptomycetaceae_refseq_30062023.tsv\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "88d2228e-1020-4ab7-ae26-53db5d98584c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GenBank Assembly ID (Accession.version)</th>\n",
       "      <th>GenBank release ID</th>\n",
       "      <th>RefSeq release ID</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RefSeq Assembly ID (Accession.version)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GCF_000009765.2</th>\n",
       "      <td>GCA_000009765.2</td>\n",
       "      <td>3140618</td>\n",
       "      <td>3192468</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000010605.1</th>\n",
       "      <td>GCA_000010605.1</td>\n",
       "      <td>11408</td>\n",
       "      <td>32088</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000091305.1</th>\n",
       "      <td>GCA_000091305.1</td>\n",
       "      <td>111938</td>\n",
       "      <td>163818</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000092385.1</th>\n",
       "      <td>GCA_000092385.1</td>\n",
       "      <td>113018</td>\n",
       "      <td>354548</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000147815.2</th>\n",
       "      <td>GCA_000147815.3</td>\n",
       "      <td>300768</td>\n",
       "      <td>301138</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       GenBank Assembly ID (Accession.version)  \\\n",
       "RefSeq Assembly ID (Accession.version)                                           \n",
       "GCF_000009765.2                                                GCA_000009765.2   \n",
       "GCF_000010605.1                                                GCA_000010605.1   \n",
       "GCF_000091305.1                                                GCA_000091305.1   \n",
       "GCF_000092385.1                                                GCA_000092385.1   \n",
       "GCF_000147815.2                                                GCA_000147815.3   \n",
       "\n",
       "                                        GenBank release ID  \\\n",
       "RefSeq Assembly ID (Accession.version)                       \n",
       "GCF_000009765.2                                    3140618   \n",
       "GCF_000010605.1                                      11408   \n",
       "GCF_000091305.1                                     111938   \n",
       "GCF_000092385.1                                     113018   \n",
       "GCF_000147815.2                                     300768   \n",
       "\n",
       "                                        RefSeq release ID   Unnamed: 4  \n",
       "RefSeq Assembly ID (Accession.version)                                  \n",
       "GCF_000009765.2                                    3192468         NaN  \n",
       "GCF_000010605.1                                      32088         NaN  \n",
       "GCF_000091305.1                                     163818         NaN  \n",
       "GCF_000092385.1                                     354548         NaN  \n",
       "GCF_000147815.2                                     301138         NaN  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ncbi_accn = pd.read_csv(ncbi_accn_table, sep=\"\\t\", index_col=2)\n",
    "df_ncbi_accn.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "312b6d2c-57ef-4e5c-9349-8449836f5585",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source</th>\n",
       "      <th>organism</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "      <th>strain</th>\n",
       "      <th>closest_placement_reference</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genome_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GCF_000009765.2</th>\n",
       "      <td>ncbi</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000010605.1</th>\n",
       "      <td>ncbi</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000091305.1</th>\n",
       "      <td>ncbi</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000092385.1</th>\n",
       "      <td>ncbi</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000147815.2</th>\n",
       "      <td>ncbi</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                source organism genus species strain  \\\n",
       "genome_id                                              \n",
       "GCF_000009765.2   ncbi      NaN   NaN     NaN    NaN   \n",
       "GCF_000010605.1   ncbi      NaN   NaN     NaN    NaN   \n",
       "GCF_000091305.1   ncbi      NaN   NaN     NaN    NaN   \n",
       "GCF_000092385.1   ncbi      NaN   NaN     NaN    NaN   \n",
       "GCF_000147815.2   ncbi      NaN   NaN     NaN    NaN   \n",
       "\n",
       "                closest_placement_reference  \n",
       "genome_id                                    \n",
       "GCF_000009765.2                         NaN  \n",
       "GCF_000010605.1                         NaN  \n",
       "GCF_000091305.1                         NaN  \n",
       "GCF_000092385.1                         NaN  \n",
       "GCF_000147815.2                         NaN  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "samples_columns = [\"source\",\"organism\",\"genus\",\"species\",\"strain\",\"closest_placement_reference\"]\n",
    "df_samples_strepto_ncbi_all = pd.DataFrame(index=df_ncbi_accn.index, columns=samples_columns)\n",
    "df_samples_strepto_ncbi_all.index.name = \"genome_id\"\n",
    "df_samples_strepto_ncbi_all.source = \"ncbi\"\n",
    "df_samples_strepto_ncbi_all.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "bc3a0e12-abbf-4c03-bfd4-52dbe7879d21",
   "metadata": {},
   "outputs": [],
   "source": [
    "# df_samples_strepto_ncbi_all.to_csv(tables_dir / \"df_samples_strepto_ncbi_all.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5a55e86c-aa7e-45fd-b7bc-6042e2818d67",
   "metadata": {},
   "source": [
    "### Setup the config project using BGCFlow wrapper\n",
    "\n",
    "Follow the guide to set up a project called \"qc_strepto_ncbi\" with about df_samples tables and project config with the following rules\n",
    "\n",
    "```\n",
    "cd bgcflow_dir\n",
    "bgcflow init --project qc_strepto_ncbi\n",
    "cp .examples/_genome_project_example/prokka-db.csv config/qc_strepto_ncbi/\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "bb9c9dce-20fe-4e6c-99fd-ce6731167069",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Write samples.csv table to config directory of qc_strepto_ncbi project\n",
    "bgcflow_dir = Path(notebook_configuration[\"bgcflow_dir\"])\n",
    "project_name_1 = \"qc_strepto_ncbi\"\n",
    "\n",
    "config_dir_1 = bgcflow_dir / f\"config/{project_name_1}\"\n",
    "# df_samples.to_csv(config_dir / \"samples.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aabd6101-d891-4400-ba38-bf7148dd95b8",
   "metadata": {},
   "source": [
    "## Update project_config\n",
    "\n",
    "Update the file `config/qc_strepto_ncbi/project_config.yaml` to allow seqfu rule on.\n",
    "\n",
    "Note that the latest release 214 of GTDB was used here with version 2.3.0 updated in the `workflow/envs/gtdbtk.yaml` file.\n",
    "\n",
    "```\n",
    "name: qc_strepto_ncbi\n",
    "pep_version: 2.1.0\n",
    "description: '2938 genome accessions were collected from the NCBI RefSeq database \n",
    "on the 30-06-2023 for genomes belonging to Streptomycetaceae family'\n",
    "sample_table: samples.csv\n",
    "prokka-db: 'prokka-db.csv'\n",
    "rules:\n",
    "  seqfu: true\n",
    "\n",
    "#### CUSTOM RULE CONFIGURATION ####\n",
    "rule_parameters:  install_gtdbtk:\n",
    "    release: \"214\"\n",
    "    release_version: \"214\"\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "61b3fef4-9946-4317-8077-01766caef602",
   "metadata": {},
   "source": [
    "## Evaluate the results of the above first BGCFlow run\n",
    "\n",
    "Key tables generated:\n",
    "\n",
    "1. df_ncbi_meta : NCBI metadata\n",
    "2. df_gtdb_meta: GTDB metadata for genomes available in the release 214\n",
    "3. df_seqfu_meta: Genome assembly statistics using seqfu rule"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "0dbc4e7d-30c6-4649-a873-bf07eb84b3a7",
   "metadata": {},
   "outputs": [],
   "source": [
    "processed_dir_1 = bgcflow_dir / \"data\" / \"processed\" / project_name_1\n",
    "\n",
    "# Read output tables from the processed directory\n",
    "ncbi_meta_table_1 = processed_dir_1 / \"tables\"/ \"df_ncbi_meta.csv\"\n",
    "df_ncbi_meta_1 = pd.read_csv(ncbi_meta_table_1, index_col= 0)\n",
    "\n",
    "gtdb_meta_table_1 = processed_dir_1 / \"tables\"/ \"df_gtdb_meta.csv\"\n",
    "df_gtdb_meta_1 = pd.read_csv(gtdb_meta_table_1, index_col= 0)\n",
    "\n",
    "seqfu_meta_table_1 = processed_dir_1 / \"tables\"/ \"df_seqfu_stats.csv\"\n",
    "df_seqfu_meta_1 = pd.read_csv(seqfu_meta_table_1, index_col= 0)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3c105562-57f2-4a50-a92c-ceb5702e4d37",
   "metadata": {},
   "source": [
    "### Genomes not found in GTDB \n",
    "\n",
    "A particular version of GTDB might not contain all of the genomes from NCBI. Some genomes dont have information on the CheckM. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "86f1d2b1-0502-46cc-b087-0d19c791960b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "detail\n",
       "Genome found                                                  2542\n",
       "Genome not found                                               390\n",
       "Genome not found - no taxonomic assignment in release R214       6\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gtdb_meta_1.detail.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "193b7a2f-d9f6-4c96-89a4-d82ee441a408",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Genus\n",
       "g__Streptomyces           2468\n",
       "g__                        364\n",
       "g__Kitasatospora            70\n",
       "g__Streptacidiphilus        15\n",
       "g__Embleya                   4\n",
       "g__Streptomyces_D            3\n",
       "g__Streptomyces_A            3\n",
       "g__Yinghuangia               2\n",
       "g__Amycolatopsis             1\n",
       "g__Streptacidiphilus_A       1\n",
       "g__Williamsia                1\n",
       "g__Quadrisphaera             1\n",
       "g__Streptomyces_G            1\n",
       "g__Kribbella                 1\n",
       "g__Allostreptomyces          1\n",
       "g__SCUT-3                    1\n",
       "g__Moraxella_A               1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gtdb_meta_1.Genus.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "9583576c-0006-4b86-8b8e-a8af318949ed",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df_gtdb_absent = df_gtdb_meta_1[df_gtdb_meta_1.detail != \"Genome found\"]\n",
    "gtdb_absent_table = processed_dir_1 / \"tables\"/ \"df_gtdb_absent.csv\"\n",
    "\n",
    "# df_gtdb_absent.to_csv(gtdb_absent_table)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "436d6cf9-354c-4b9e-a41e-6b1b90a9b012",
   "metadata": {},
   "source": [
    "# 2. Set up GTDBtk run for the remaining genomes\n",
    "\n",
    "In this second BGCFlow run, a project is setup to run GTDBTk and CheckM to assess the taxonomy and assembly quality for genomes not found in R214 version of GTDB server along with the genomes from the NBC strain collection.\n",
    "\n",
    "1. df_gtdb_absent : 396 NCBI genomes that were not found in the GTDB server.\n",
    "2. df_samples_NBC_1023: List of 1023 actinomycetes from the NBC collection. Note that some of these are redundant with NCBI database and thus are removed."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0c16f3c8-6feb-4a54-be64-4cc9a08634ea",
   "metadata": {},
   "source": [
    "### Setup the config project using BGCFlow wrapper\n",
    "\n",
    "Follow the guide to set up a project called \"qc_gtdbtk\" with samples tables and project config with the following rules\n",
    "\n",
    "```\n",
    "cd bgcflow_dir\n",
    "bgcflow init --project qc_gtdbtk\n",
    "cp .examples/_genome_project_example/prokka-db.csv config/qc_gtdbtk/\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3ee75197-5f1f-4b8b-97a8-6312c8f060a2",
   "metadata": {},
   "source": [
    "## Update project_config\n",
    "\n",
    "Update the file `config/qc_strepto_ncbi/project_config.yaml` to allow seqfu rule on.\n",
    "\n",
    "Note that the latest release 214 of GTDB was used here with version 2.3.0 updated in the `workflow/envs/gtdbtk.yaml` file.\n",
    "\n",
    "```\n",
    "name: qc_gtdbtk\n",
    "pep_version: 2.1.0\n",
    "description: 'Project to assess taxonomy and assembly quality of 396 NCBI + 902 NBC genomes.'\n",
    "sample_table: samples.csv\n",
    "prokka-db: 'prokka-db.csv'\n",
    "rules:\n",
    "  seqfu: true\n",
    "  checkm: false\n",
    "  gtdbtk: false\n",
    "\n",
    "#### CUSTOM RULE CONFIGURATION ####\n",
    "rule_p  install_gtdbtk:\n",
    "    release: \"214\"lease: \"214\"\n",
    "    release_version: \"214\"\n",
    "\"```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "926e9049-3d7d-47b5-b5c0-8ad4c75b8efa",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Write samples.csv table to config directory of qc_strepto_ncbi project\n",
    "project_name_2 = \"qc_gtdbtk\"\n",
    "\n",
    "config_dir_2 = bgcflow_dir / f\"config/{project_name_2}\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "e9e7a4c7-9304-4ef3-9a59-aee4798e2c24",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read NBC data file (manually created for custom genomes). Follow documentation of BGCFlow for custom samples.\n",
    "samples_NBC_table = config_dir_2 / \"samples_NBC_1023.csv\"\n",
    "df_samples_NBC_1023 = pd.read_csv(samples_NBC_table, index_col=0)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "28e6704c-bd0a-45f3-b2b9-362d64d1079d",
   "metadata": {},
   "source": [
    "### Remove redundant genomes\n",
    "\n",
    "As some of th NBC genomes were already part of NCBI dataset on 30 June 2023, we removed them from the custom project and kept the NCBI versions. All these genomes were part of the BioProject: PRJNA747871."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "fb8a9f69-fd48-4021-91d5-da4405f04931",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The number of genomes that were removed from NBC dataset: 121\n"
     ]
    }
   ],
   "source": [
    "redundant_strain_list = df_ncbi_meta_1[df_ncbi_meta_1.BioProject == \"PRJNA747871\"].strain.tolist()\n",
    "print(\"The number of genomes that were removed from NBC dataset:\", len(redundant_strain_list))\n",
    "df_samples_NBC_902 = df_samples_NBC_1023.drop(redundant_strain_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "ac641f80-699b-4fe7-b0d0-1a80d7e5b1d0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "assembly_level\n",
       "Contig             1495\n",
       "Scaffold            936\n",
       "Complete Genome     453\n",
       "Chromosome           54\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ncbi_meta_1.assembly_level.value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "96ac14b0-7add-427d-98ad-d980c634e6c2",
   "metadata": {},
   "source": [
    "### Add fasta files for custom genomes\n",
    "\n",
    "The fasta files for all 902 genomes were manuaully added to the directory `data/raw/fasta/`."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f5c3c11c-7507-43b4-af3b-582c5a880758",
   "metadata": {},
   "source": [
    "### Combine the GTDBTk project \n",
    "\n",
    "Samples table for the GTDBTk project is created with the 902 NBC genomes and 396 NCBI genomes without taxonomic/ assembly quality informatio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "335aeb4a-1d57-4f42-9e3b-f94e530b78fd",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_samples_gtdb_absent = df_samples_strepto_ncbi_all.loc[df_gtdb_absent.index,:] \n",
    "\n",
    "df_samples_gtdbtk = pd.concat([df_samples_NBC_902, df_samples_gtdb_absent])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "92c00b3e-ec1d-4351-b5e5-3a1e24af18cb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Save the samples tables\n",
    "# df_samples_gtdb_absent.to_csv(config_dir_2/ \"samples_NCBI_gtdb_absent.csv\")\n",
    "# df_samples_NBC_902.to_csv(config_dir_2/ \"samples_NBC_902.csv\")\n",
    "# df_samples_gtdbtk.to_csv(config_dir_2/ \"samples.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f4111f3c-8228-4141-8aa1-e899a733c662",
   "metadata": {},
   "source": [
    "## Evaluate the results of the above first BGCFlow run on gtdbtk\n",
    "\n",
    "Key tables generated:\n",
    "\n",
    "1. gtdbtk.bac120.summary.tsv: GTDB-Tk based taxonomic identification with version 214\n",
    "2. df_seqfu_stats: Genome assembly statistics using seqfu rule\n",
    "3. df_checkm_stats: Genome assembly statistics using CheckM rule"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "f7cf6a32-25ae-484c-8f59-46853f84048b",
   "metadata": {},
   "outputs": [],
   "source": [
    "processed_dir_2 = bgcflow_dir / \"data\" / \"processed\" / project_name_2\n",
    "\n",
    "# Read output tables from the processed directory\n",
    "ncbi_meta_table_2 = processed_dir_2 / \"tables\"/ \"df_ncbi_meta.csv\"\n",
    "df_ncbi_meta_2 = pd.read_csv(ncbi_meta_table_2, index_col= 0)\n",
    "\n",
    "gtdb_meta_table_2 = processed_dir_2 / \"tables\"/ \"df_gtdb_meta.csv\"\n",
    "df_gtdb_meta_2 = pd.read_csv(gtdb_meta_table_2, index_col= 0)\n",
    "\n",
    "seqfu_meta_table_2 = processed_dir_2 / \"tables\"/ \"df_seqfu_stats.csv\"\n",
    "df_seqfu_meta_2 = pd.read_csv(seqfu_meta_table_2, index_col= 0)\n",
    "\n",
    "check_meta_table_2 = processed_dir_2 / \"tables\"/ \"df_checkm_stats.csv\"\n",
    "df_checkm_meta_2 = pd.read_csv(check_meta_table_2, index_col= 0)\n",
    "\n",
    "gtdbtk_meta_table_2 = processed_dir_2 / \"tables\"/ \"gtdbtk.bac120.summary.tsv\"\n",
    "df_gtdbtk_meta_2 = pd.read_csv(gtdbtk_meta_table_2, index_col= 0, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "f461054c-efdd-4f77-a336-93c854360060",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th>GC</th>\n",
       "      <th>GC std</th>\n",
       "      <th>Genome size</th>\n",
       "      <th># ambiguous bases</th>\n",
       "      <th># scaffolds</th>\n",
       "      <th># contigs</th>\n",
       "      <th>Longest scaffold</th>\n",
       "      <th>Longest contig</th>\n",
       "      <th>N50 (scaffolds)</th>\n",
       "      <th>N50 (contigs)</th>\n",
       "      <th>Mean scaffold length</th>\n",
       "      <th>Mean contig length</th>\n",
       "      <th>Coding density</th>\n",
       "      <th>Translation table</th>\n",
       "      <th># predicted genes</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genome_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NBC_01635</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.710487</td>\n",
       "      <td>0.030986</td>\n",
       "      <td>9003324</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>8982372</td>\n",
       "      <td>8982372</td>\n",
       "      <td>8982372</td>\n",
       "      <td>8982372</td>\n",
       "      <td>4.501662e+06</td>\n",
       "      <td>4.501662e+06</td>\n",
       "      <td>0.897961</td>\n",
       "      <td>4</td>\n",
       "      <td>3814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_013302945.1</th>\n",
       "      <td>8.620690</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.713604</td>\n",
       "      <td>0.028444</td>\n",
       "      <td>1335511</td>\n",
       "      <td>10254</td>\n",
       "      <td>332</td>\n",
       "      <td>443</td>\n",
       "      <td>61290</td>\n",
       "      <td>56941</td>\n",
       "      <td>11311</td>\n",
       "      <td>8543</td>\n",
       "      <td>4.022623e+03</td>\n",
       "      <td>2.991551e+03</td>\n",
       "      <td>0.829939</td>\n",
       "      <td>11</td>\n",
       "      <td>1446</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_023669525.1</th>\n",
       "      <td>82.445141</td>\n",
       "      <td>60.736677</td>\n",
       "      <td>0.716594</td>\n",
       "      <td>0.010725</td>\n",
       "      <td>14173392</td>\n",
       "      <td>0</td>\n",
       "      <td>182</td>\n",
       "      <td>182</td>\n",
       "      <td>404509</td>\n",
       "      <td>404509</td>\n",
       "      <td>109952</td>\n",
       "      <td>109952</td>\n",
       "      <td>7.787578e+04</td>\n",
       "      <td>7.787578e+04</td>\n",
       "      <td>0.890316</td>\n",
       "      <td>11</td>\n",
       "      <td>13000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_025398775.1</th>\n",
       "      <td>90.665236</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.724229</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>7687527</td>\n",
       "      <td>222271</td>\n",
       "      <td>1</td>\n",
       "      <td>1020</td>\n",
       "      <td>7687527</td>\n",
       "      <td>70481</td>\n",
       "      <td>7687527</td>\n",
       "      <td>11827</td>\n",
       "      <td>7.687527e+06</td>\n",
       "      <td>7.318878e+03</td>\n",
       "      <td>0.817609</td>\n",
       "      <td>11</td>\n",
       "      <td>6999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_016432765.1</th>\n",
       "      <td>91.244635</td>\n",
       "      <td>0.085837</td>\n",
       "      <td>0.721482</td>\n",
       "      <td>0.042171</td>\n",
       "      <td>8364241</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>8111009</td>\n",
       "      <td>8111009</td>\n",
       "      <td>8111009</td>\n",
       "      <td>8111009</td>\n",
       "      <td>1.672848e+06</td>\n",
       "      <td>1.672848e+06</td>\n",
       "      <td>0.834695</td>\n",
       "      <td>11</td>\n",
       "      <td>7920</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_007677905.1</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>12.970457</td>\n",
       "      <td>0.720057</td>\n",
       "      <td>0.026564</td>\n",
       "      <td>8192675</td>\n",
       "      <td>920</td>\n",
       "      <td>1578</td>\n",
       "      <td>1591</td>\n",
       "      <td>757584</td>\n",
       "      <td>442105</td>\n",
       "      <td>224336</td>\n",
       "      <td>195329</td>\n",
       "      <td>5.191809e+03</td>\n",
       "      <td>5.148809e+03</td>\n",
       "      <td>0.873614</td>\n",
       "      <td>11</td>\n",
       "      <td>8485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NBC_01801</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>3.479466</td>\n",
       "      <td>0.710333</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>11975494</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>11975494</td>\n",
       "      <td>11975494</td>\n",
       "      <td>11975494</td>\n",
       "      <td>11975494</td>\n",
       "      <td>1.197549e+07</td>\n",
       "      <td>1.197549e+07</td>\n",
       "      <td>0.890382</td>\n",
       "      <td>11</td>\n",
       "      <td>10987</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_001636945.2</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>1.394850</td>\n",
       "      <td>0.710039</td>\n",
       "      <td>0.013189</td>\n",
       "      <td>10017507</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>9260562</td>\n",
       "      <td>9260562</td>\n",
       "      <td>9260562</td>\n",
       "      <td>9260562</td>\n",
       "      <td>3.339169e+06</td>\n",
       "      <td>3.339169e+06</td>\n",
       "      <td>0.865050</td>\n",
       "      <td>11</td>\n",
       "      <td>8924</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_001279805.1</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>1.754386</td>\n",
       "      <td>0.734291</td>\n",
       "      <td>0.024386</td>\n",
       "      <td>9490701</td>\n",
       "      <td>0</td>\n",
       "      <td>528</td>\n",
       "      <td>528</td>\n",
       "      <td>212011</td>\n",
       "      <td>212011</td>\n",
       "      <td>46679</td>\n",
       "      <td>46679</td>\n",
       "      <td>1.797481e+04</td>\n",
       "      <td>1.797481e+04</td>\n",
       "      <td>0.867294</td>\n",
       "      <td>11</td>\n",
       "      <td>8328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NBC_01796</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>1.315789</td>\n",
       "      <td>0.705956</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>8562983</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>8562983</td>\n",
       "      <td>8562983</td>\n",
       "      <td>8562983</td>\n",
       "      <td>8562983</td>\n",
       "      <td>8.562983e+06</td>\n",
       "      <td>8.562983e+06</td>\n",
       "      <td>0.889024</td>\n",
       "      <td>11</td>\n",
       "      <td>7518</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1298 rows Ã— 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Completeness  Contamination        GC    GC std  Genome size  \\\n",
       "genome_id                                                                       \n",
       "NBC_01635            0.000000       0.000000  0.710487  0.030986      9003324   \n",
       "GCF_013302945.1      8.620690       0.000000  0.713604  0.028444      1335511   \n",
       "GCF_023669525.1     82.445141      60.736677  0.716594  0.010725     14173392   \n",
       "GCF_025398775.1     90.665236       0.000000  0.724229  0.000000      7687527   \n",
       "GCF_016432765.1     91.244635       0.085837  0.721482  0.042171      8364241   \n",
       "...                       ...            ...       ...       ...          ...   \n",
       "GCF_007677905.1    100.000000      12.970457  0.720057  0.026564      8192675   \n",
       "NBC_01801          100.000000       3.479466  0.710333  0.000000     11975494   \n",
       "GCF_001636945.2    100.000000       1.394850  0.710039  0.013189     10017507   \n",
       "GCF_001279805.1    100.000000       1.754386  0.734291  0.024386      9490701   \n",
       "NBC_01796          100.000000       1.315789  0.705956  0.000000      8562983   \n",
       "\n",
       "                 # ambiguous bases  # scaffolds  # contigs  Longest scaffold  \\\n",
       "genome_id                                                                      \n",
       "NBC_01635                        0            2          2           8982372   \n",
       "GCF_013302945.1              10254          332        443             61290   \n",
       "GCF_023669525.1                  0          182        182            404509   \n",
       "GCF_025398775.1             222271            1       1020           7687527   \n",
       "GCF_016432765.1                  0            5          5           8111009   \n",
       "...                            ...          ...        ...               ...   \n",
       "GCF_007677905.1                920         1578       1591            757584   \n",
       "NBC_01801                        0            1          1          11975494   \n",
       "GCF_001636945.2                  0            3          3           9260562   \n",
       "GCF_001279805.1                  0          528        528            212011   \n",
       "NBC_01796                        0            1          1           8562983   \n",
       "\n",
       "                 Longest contig  N50 (scaffolds)  N50 (contigs)  \\\n",
       "genome_id                                                         \n",
       "NBC_01635               8982372          8982372        8982372   \n",
       "GCF_013302945.1           56941            11311           8543   \n",
       "GCF_023669525.1          404509           109952         109952   \n",
       "GCF_025398775.1           70481          7687527          11827   \n",
       "GCF_016432765.1         8111009          8111009        8111009   \n",
       "...                         ...              ...            ...   \n",
       "GCF_007677905.1          442105           224336         195329   \n",
       "NBC_01801              11975494         11975494       11975494   \n",
       "GCF_001636945.2         9260562          9260562        9260562   \n",
       "GCF_001279805.1          212011            46679          46679   \n",
       "NBC_01796               8562983          8562983        8562983   \n",
       "\n",
       "                 Mean scaffold length  Mean contig length  Coding density  \\\n",
       "genome_id                                                                   \n",
       "NBC_01635                4.501662e+06        4.501662e+06        0.897961   \n",
       "GCF_013302945.1          4.022623e+03        2.991551e+03        0.829939   \n",
       "GCF_023669525.1          7.787578e+04        7.787578e+04        0.890316   \n",
       "GCF_025398775.1          7.687527e+06        7.318878e+03        0.817609   \n",
       "GCF_016432765.1          1.672848e+06        1.672848e+06        0.834695   \n",
       "...                               ...                 ...             ...   \n",
       "GCF_007677905.1          5.191809e+03        5.148809e+03        0.873614   \n",
       "NBC_01801                1.197549e+07        1.197549e+07        0.890382   \n",
       "GCF_001636945.2          3.339169e+06        3.339169e+06        0.865050   \n",
       "GCF_001279805.1          1.797481e+04        1.797481e+04        0.867294   \n",
       "NBC_01796                8.562983e+06        8.562983e+06        0.889024   \n",
       "\n",
       "                 Translation table  # predicted genes  \n",
       "genome_id                                              \n",
       "NBC_01635                        4               3814  \n",
       "GCF_013302945.1                 11               1446  \n",
       "GCF_023669525.1                 11              13000  \n",
       "GCF_025398775.1                 11               6999  \n",
       "GCF_016432765.1                 11               7920  \n",
       "...                            ...                ...  \n",
       "GCF_007677905.1                 11               8485  \n",
       "NBC_01801                       11              10987  \n",
       "GCF_001636945.2                 11               8924  \n",
       "GCF_001279805.1                 11               8328  \n",
       "NBC_01796                       11               7518  \n",
       "\n",
       "[1298 rows x 17 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_checkm_meta_2.sort_values(\"Completeness\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b4bf72ec-6d28-4738-92a6-bda0473db5e7",
   "metadata": {},
   "source": [
    "# 3. Rerun checkM on NBC_01635\n",
    "\n",
    "Above assessment showed that CheckM registered the completeness value of 0 for the genome NBC_01635. Thus, we reran CheckM only on this genome. \n",
    "\n",
    "NOTE: It seemed that translation table 4 was being used instead of 11. We manually edited the rule of checkM to use custom Prokka based gene annotation as input instead of fasta nucleotide file. "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "53760f87-ed61-4f66-b294-5d65320371f4",
   "metadata": {},
   "source": [
    "### Setup the config project using BGCFlow wrapper\n",
    "\n",
    "Follow the guide to set up a project called \"checkm_rerun\" with samples tables and project config with the following rules\n",
    "\n",
    "```\n",
    "cd bgcflow_dir\n",
    "bgcflow init --project checkm_rerun\n",
    "cp .examples/_genome_project_example/prokka-db.csv config/checkm_rerun/\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "92778f80-f964-4b6b-a45a-c4d9e3f2f027",
   "metadata": {},
   "source": [
    "## Update project_config\n",
    "\n",
    "Update the file `config/checkm_rerun/project_config.yaml` to allow seqfu rule on.\n",
    "\n",
    "```\n",
    "name: qc_strepto_ncbi\n",
    "pep_version: 2.1.0\n",
    "description: 'CheckM rerun for the genome NBC_01635'\n",
    "sample_table: samples.csv\n",
    "prokka-db: 'prokka-db.csv'\n",
    "rules:\n",
    "  checkm: True\n",
    "```\n",
    "\n",
    "NOTE: We manually edited the rule of checkM to use custom Prokka based gene annotation as input instead of fasta nucleotide file. Thus the results will not be generated automatically.*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "067d4e3f-5656-4894-aab6-465cb31ce322",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Write samples.csv table to config directory of qc_strepto_ncbi project\n",
    "project_name_3 = \"checkm_rerun\"\n",
    "\n",
    "config_dir_3 = bgcflow_dir / f\"config/{project_name_3}\"\n",
    "\n",
    "samples_columns = [\"source\",\"organism\",\"genus\",\"species\",\"strain\",\"closest_placement_reference\"]\n",
    "df_samples_checkm_rerun = pd.DataFrame(index=[\"NBC_01635\"], columns=samples_columns)\n",
    "df_samples_checkm_rerun.index.name = \"genome_id\"\n",
    "df_samples_checkm_rerun.loc[\"NBC_01635\", :] = df_samples_gtdbtk.loc[\"NBC_01635\", :]\n",
    "df_samples_checkm_rerun.to_csv(config_dir_3 / \"samples.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "cdffab9f-a28c-4362-bb40-1fe3c877b6ed",
   "metadata": {},
   "outputs": [],
   "source": [
    "processed_dir_3 = bgcflow_dir / \"data\" / \"processed\" / project_name_3\n",
    "\n",
    "check_meta_table_3 = processed_dir_3 / \"tables\"/ \"df_checkm_stats.csv\"\n",
    "df_checkm_meta_3 = pd.read_csv(check_meta_table_3, index_col= 0)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "11e9d562-9656-47e2-9c6b-6913d2c3f0ad",
   "metadata": {},
   "source": [
    "### Update the checkM table to include the updated results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "7791659c-79a4-4741-b5df-90de73637630",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_checkm_meta_2.loc[\"NBC_01635\", :] = df_checkm_meta_3.loc[\"NBC_01635\", :]\n",
    "# df_checkm_meta_2.to_csv(processed_dir_2 / \"tables\" / \"df_checkm_stats_curated.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d6d54270-9b68-4b95-8df5-6a1bc1bee12b",
   "metadata": {},
   "source": [
    "### Assess gtdbtk data and extract Streptomyces genus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "8222178f-dd42-4afd-a46b-9c46adebc681",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove genomes without classification\n",
    "df_gtdb_meta = df_gtdb_meta_1.copy()\n",
    "df_gtdbtk_meta_2 = df_gtdbtk_meta_2[df_gtdbtk_meta_2.classification != \"Unclassified Bacteria\"]\n",
    "\n",
    "tax_mapping = {\"d\" : \"Domain\",\n",
    "              \"p\" : \"Phylum\",\n",
    "              \"c\" : \"Class\",\n",
    "              \"o\" : \"Order\",\n",
    "              \"f\" : \"Family\",\n",
    "              \"g\" : \"Genus\",\n",
    "              \"s\" : \"Organism\"}\n",
    "\n",
    "for index in df_gtdbtk_meta_2.index:\n",
    "    tax = [i for i in df_gtdbtk_meta_2.loc[index, \"classification\"].split(\";\")]\n",
    "    for level in tax:\n",
    "        key = level.split(\"__\")[0]\n",
    "        if key == \"g\":\n",
    "            genus = level.split(\"__\")[-1]\n",
    "        if level == \"s__\":\n",
    "            level = f\"s__{genus} sp.\"\n",
    "        df_gtdb_meta.loc[index, tax_mapping[key]] = level\n",
    "    df_gtdb_meta.loc[index, \"gtdb_release\"] = \"R214\"\n",
    "    df_gtdb_meta.loc[index, \"detail\"] = \"found using gtdbtk\"\n",
    "df_gtdb_meta.Species = [i.split()[-1] for i in df_gtdb_meta.Organism]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "e957df26-98db-454d-90a0-ae3109da8310",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_gtdbtk_meta_curated_2 = df_gtdb_meta.loc[df_seqfu_meta_2.index,:]\n",
    "# df_gtdbtk_meta_curated_2.to_csv(processed_dir_2/ \"tables\"/ \"df_gtdbtk_meta_curated.csv\")\n",
    "# df_gtdb_meta.to_csv(processed_dir_1/ \"tables\"/ \"df_gtdb_meta_curated.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "427b2e0a-996b-4034-acc1-9370761c5e05",
   "metadata": {},
   "source": [
    "### Assess checkM, SeqFu data and define quality filters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "e12f9e4e-ec54-4637-aac1-6d89c32e8682",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_seqfu_meta = df_seqfu_meta_1.combine_first(df_seqfu_meta_2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "e3a61d0f-3a78-4894-90f2-fcf6be8ee067",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Manually load NBC genomes assembly_level status\n",
    "df_nbc_assembly_level = pd.read_csv(config_dir_2/ \"df_nbc_assembly_level.csv\", index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "4795249a-e181-45f2-b38e-0dacbfc2437e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCF_000154885.1 removed. Contamination: 2.39 Completeness: 84.47\n",
      "GCF_000154925.1 removed. Contamination: 1.4 Completeness: 75.3\n",
      "GCF_000179195.1 removed. Contigs: 254 N50: 59002\n",
      "GCF_000179215.1 removed. Contigs: 1275 N50: 9224\n",
      "GCF_000225525.1 removed. Contigs: 525 N50: 31811\n",
      "GCF_000239075.1 removed. Contigs: 284 N50: 59300\n",
      "GCF_000242715.1 removed. Contigs: 244 N50: 113361\n",
      "GCF_000258595.1 removed. Contigs: 243 N50: 33216\n",
      "GCF_000259895.1 removed. Contigs: 729 N50: 20147\n",
      "GCF_000262345.1 removed. Contigs: 385 N50: 30588\n",
      "GCF_000280905.1 removed. Contigs: 109 N50: 121115\n",
      "GCF_000317595.1 removed. Contigs: 687 N50: 26361\n",
      "GCF_000331005.1 removed. Contigs: 692 N50: 27448\n",
      "GCF_000332625.1 removed. Contigs: 379 N50: 56781\n",
      "GCF_000342125.1 removed. Contigs: 266 N50: 47194\n",
      "GCF_000342345.1 removed. Contigs: 171 N50: 79071\n",
      "GCF_000367365.1 removed. Contigs: 202 N50: 124739\n",
      "GCF_000372745.1 removed. Contigs: 191 N50: 180969\n",
      "GCF_000373585.1 removed. Contigs: 120 N50: 169228\n",
      "GCF_000376565.1 removed. Contigs: 178 N50: 135550\n",
      "GCF_000380165.1 removed. Contigs: 112 N50: 136252\n",
      "GCF_000383615.1 removed. Contigs: 116 N50: 235831\n",
      "GCF_000403765.2 removed. Contigs: 242 N50: 75627\n",
      "GCF_000411315.1 removed. Contamination: 16.81 Completeness: 100.0\n",
      "GCF_000414115.1 removed. Contigs: 568 N50: 32382\n",
      "GCF_000415505.1 removed. Contigs: 1726 N50: 23632\n",
      "GCF_000498935.1 removed. Contigs: 1508 N50: 6247\n",
      "GCF_000504065.1 removed. Contigs: 244 N50: 67182\n",
      "GCF_000518045.1 removed. Contigs: 367 N50: 46890\n",
      "GCF_000590515.1 removed. Contigs: 290 N50: 95768\n",
      "GCF_000696595.1 removed. Contigs: 602 N50: 27217\n",
      "GCF_000698945.1 removed. Contamination: 0.57 Completeness: 89.7\n",
      "GCF_000700005.2 removed. Contigs: 152 N50: 358441\n",
      "GCF_000707965.1 removed. Contigs: 375 N50: 419227\n",
      "GCF_000710465.2 removed. Contigs: 846 N50: 24761\n",
      "GCF_000715745.1 removed. Contigs: 130 N50: 130234\n",
      "GCF_000715845.1 removed. Contigs: 223 N50: 86123\n",
      "GCF_000716215.1 removed. Contigs: 141 N50: 141587\n",
      "GCF_000716335.1 removed. Contigs: 114 N50: 230062\n",
      "GCF_000716435.1 removed. Contigs: 211 N50: 356030\n",
      "GCF_000716445.1 removed. Contigs: 259 N50: 125151\n",
      "GCF_000716455.1 removed. Contigs: 138 N50: 167875\n",
      "GCF_000716515.1 removed. Contigs: 140 N50: 199375\n",
      "GCF_000716525.1 removed. Contigs: 334 N50: 166038\n",
      "GCF_000716535.1 removed. Contigs: 232 N50: 194920\n",
      "GCF_000716545.1 removed. Contigs: 227 N50: 89226\n",
      "GCF_000716605.1 removed. Contigs: 242 N50: 208591\n",
      "GCF_000716615.1 removed. Contigs: 408 N50: 110620\n",
      "GCF_000716625.1 removed. Contigs: 155 N50: 142934\n",
      "GCF_000716675.1 removed. Contigs: 171 N50: 134286\n",
      "GCF_000716685.1 removed. Contigs: 181 N50: 171873\n",
      "GCF_000716745.1 removed. Contigs: 134 N50: 178926\n",
      "GCF_000716805.1 removed. Contigs: 323 N50: 150519\n",
      "GCF_000716845.1 removed. Contigs: 144 N50: 178168\n",
      "GCF_000716875.1 removed. Contigs: 250 N50: 73927\n",
      "GCF_000716885.1 removed. Contigs: 505 N50: 127928\n",
      "GCF_000716935.1 removed. Contigs: 263 N50: 264758\n",
      "GCF_000716965.1 removed. Contigs: 298 N50: 97127\n",
      "GCF_000717025.1 removed. Contigs: 106 N50: 332139\n",
      "GCF_000717055.1 removed. Contigs: 181 N50: 287302\n",
      "GCF_000717105.1 removed. Contigs: 178 N50: 201367\n",
      "GCF_000717185.1 removed. Contigs: 340 N50: 69365\n",
      "GCF_000717285.1 removed. Contigs: 129 N50: 199295\n",
      "GCF_000717345.1 removed. Contigs: 131 N50: 181980\n",
      "GCF_000717375.1 removed. Contigs: 135 N50: 186085\n",
      "GCF_000717595.1 removed. Contigs: 170 N50: 235906\n",
      "GCF_000717635.1 removed. Contigs: 201 N50: 89962\n",
      "GCF_000717645.1 removed. Contigs: 105 N50: 251927\n",
      "GCF_000717655.1 removed. Contigs: 151 N50: 193099\n",
      "GCF_000717715.1 removed. Contigs: 186 N50: 140273\n",
      "GCF_000717725.1 removed. Contigs: 139 N50: 107132\n",
      "GCF_000717795.1 removed. Contigs: 166 N50: 457602\n",
      "GCF_000717805.1 removed. Contigs: 205 N50: 111583\n",
      "GCF_000717815.1 removed. Contigs: 190 N50: 105458\n",
      "GCF_000717845.1 removed. Contigs: 170 N50: 148200\n",
      "GCF_000717875.1 removed. Contigs: 120 N50: 198010\n",
      "GCF_000717895.1 removed. Contigs: 221 N50: 185581\n",
      "GCF_000717925.1 removed. Contigs: 193 N50: 176453\n",
      "GCF_000717945.1 removed. Contigs: 198 N50: 122830\n",
      "GCF_000717965.1 removed. Contigs: 126 N50: 178044\n",
      "GCF_000717995.1 removed. Contigs: 282 N50: 176436\n",
      "GCF_000718025.1 removed. Contigs: 138 N50: 241189\n",
      "GCF_000718095.1 removed. Contigs: 117 N50: 275119\n",
      "GCF_000718135.1 removed. Contigs: 127 N50: 353815\n",
      "GCF_000718155.1 removed. Contigs: 357 N50: 164157\n",
      "GCF_000718165.1 removed. Contigs: 263 N50: 163841\n",
      "GCF_000718175.1 removed. Contamination: 6.94 Completeness: 99.84\n",
      "GCF_000718235.1 removed. Contigs: 142 N50: 246180\n",
      "GCF_000718295.1 removed. Contigs: 318 N50: 140407\n",
      "GCF_000718305.1 removed. Contigs: 108 N50: 381655\n",
      "GCF_000718315.1 removed. Contigs: 244 N50: 135213\n",
      "GCF_000718375.1 removed. Contigs: 211 N50: 203395\n",
      "GCF_000718525.1 removed. Contigs: 239 N50: 69828\n",
      "GCF_000718595.1 removed. Contigs: 126 N50: 232053\n",
      "GCF_000718615.1 removed. Contigs: 107 N50: 213680\n",
      "GCF_000718625.1 removed. Contigs: 217 N50: 193158\n",
      "GCF_000718635.1 removed. Contigs: 118 N50: 268175\n",
      "GCF_000718675.1 removed. Contigs: 138 N50: 269003\n",
      "GCF_000718705.1 removed. Contigs: 119 N50: 196648\n",
      "GCF_000718715.1 removed. Contigs: 619 N50: 61988\n",
      "GCF_000718755.1 removed. Contigs: 191 N50: 112113\n",
      "GCF_000718775.1 removed. Contigs: 322 N50: 65939\n",
      "GCF_000718815.1 removed. Contigs: 187 N50: 134513\n",
      "GCF_000718835.1 removed. Contigs: 132 N50: 196548\n",
      "GCF_000718855.1 removed. Contigs: 140 N50: 185916\n",
      "GCF_000718865.1 removed. Contigs: 111 N50: 179168\n",
      "GCF_000718895.1 removed. Contigs: 137 N50: 176334\n",
      "GCF_000718915.1 removed. Contigs: 104 N50: 181139\n",
      "GCF_000718935.1 removed. Contigs: 173 N50: 179028\n",
      "GCF_000718945.1 removed. Contigs: 666 N50: 96156\n",
      "GCF_000718955.1 removed. Contigs: 142 N50: 157367\n",
      "GCF_000718985.1 removed. Contigs: 195 N50: 155964\n",
      "GCF_000719015.1 removed. Contigs: 432 N50: 158037\n",
      "GCF_000719035.1 removed. Contigs: 228 N50: 213728\n",
      "GCF_000719095.1 removed. Contigs: 144 N50: 245740\n",
      "GCF_000719175.1 removed. Contigs: 279 N50: 121651\n",
      "GCF_000719185.1 removed. Contigs: 247 N50: 100266\n",
      "GCF_000719255.1 removed. Contigs: 155 N50: 246757\n",
      "GCF_000719285.1 removed. Contigs: 183 N50: 232223\n",
      "GCF_000719335.1 removed. Contigs: 255 N50: 115936\n",
      "GCF_000719345.1 removed. Contigs: 673 N50: 132442\n",
      "GCF_000719365.1 removed. Contigs: 294 N50: 70195\n",
      "GCF_000719425.1 removed. Contigs: 163 N50: 144165\n",
      "GCF_000719435.1 removed. Contigs: 165 N50: 117486\n",
      "GCF_000719475.1 removed. Contigs: 237 N50: 83861\n",
      "GCF_000719495.1 removed. Contigs: 240 N50: 90004\n",
      "GCF_000719505.1 removed. Contigs: 180 N50: 353313\n",
      "GCF_000719555.1 removed. Contigs: 187 N50: 102805\n",
      "GCF_000719575.1 removed. Contigs: 215 N50: 180744\n",
      "GCF_000719605.1 removed. Contigs: 124 N50: 205734\n",
      "GCF_000719625.1 removed. Contigs: 146 N50: 149483\n",
      "GCF_000719655.1 removed. Contigs: 146 N50: 125728\n",
      "GCF_000719665.1 removed. Contigs: 183 N50: 209060\n",
      "GCF_000719705.1 removed. Contigs: 253 N50: 124046\n",
      "GCF_000719735.1 removed. Contigs: 154 N50: 161668\n",
      "GCF_000719745.1 removed. Contigs: 188 N50: 97062\n",
      "GCF_000719775.1 removed. Contigs: 117 N50: 335080\n",
      "GCF_000719785.1 removed. Contigs: 169 N50: 170501\n",
      "GCF_000719815.1 removed. Contigs: 229 N50: 120967\n",
      "GCF_000719855.1 removed. Contigs: 131 N50: 150827\n",
      "GCF_000719865.1 removed. Contigs: 126 N50: 218508\n",
      "GCF_000719925.1 removed. Contigs: 104 N50: 230623\n",
      "GCF_000719955.1 removed. Contigs: 115 N50: 160497\n",
      "GCF_000720055.1 removed. Contigs: 101 N50: 280718\n",
      "GCF_000720065.1 removed. Contigs: 568 N50: 202225\n",
      "GCF_000720115.1 removed. Contigs: 156 N50: 187787\n",
      "GCF_000720135.1 removed. Contigs: 235 N50: 270940\n",
      "GCF_000720145.1 removed. Contigs: 208 N50: 168927\n",
      "GCF_000720175.1 removed. Contigs: 228 N50: 98196\n",
      "GCF_000720185.1 removed. Contigs: 162 N50: 132248\n",
      "GCF_000720215.1 removed. Contigs: 155 N50: 298601\n",
      "GCF_000720235.1 removed. Contigs: 113 N50: 207938\n",
      "GCF_000720255.1 removed. Contigs: 265 N50: 183856\n",
      "GCF_000720295.1 removed. Contigs: 1030 N50: 222074\n",
      "GCF_000720325.1 removed. Contigs: 362 N50: 132760\n",
      "GCF_000720455.1 removed. Contigs: 133 N50: 284155\n",
      "GCF_000720475.1 removed. Contigs: 223 N50: 156070\n",
      "GCF_000720485.1 removed. Contigs: 147 N50: 211967\n",
      "GCF_000720505.1 removed. Contigs: 1504 N50: 15145\n",
      "GCF_000720565.1 removed. Contigs: 129 N50: 198600\n",
      "GCF_000720595.1 removed. Contamination: 2.31 Completeness: 87.9\n",
      "GCF_000720605.1 removed. Contigs: 129 N50: 196306\n",
      "GCF_000720635.1 removed. Contigs: 233 N50: 91579\n",
      "GCF_000720645.1 removed. Contigs: 252 N50: 98420\n",
      "GCF_000720675.1 removed. Contigs: 130 N50: 184506\n",
      "GCF_000720685.1 removed. Contigs: 119 N50: 180248\n",
      "GCF_000720715.1 removed. Contigs: 214 N50: 100330\n",
      "GCF_000720725.1 removed. Contigs: 292 N50: 78793\n",
      "GCF_000720755.1 removed. Contigs: 236 N50: 154951\n",
      "GCF_000720765.1 removed. Contigs: 163 N50: 508937\n",
      "GCF_000720795.1 removed. Contigs: 129 N50: 167750\n",
      "GCF_000720835.1 removed. Contigs: 163 N50: 185677\n",
      "GCF_000720875.1 removed. Contigs: 271 N50: 114732\n",
      "GCF_000721035.1 removed. Contigs: 174 N50: 95686\n",
      "GCF_000721045.1 removed. Contigs: 183 N50: 169596\n",
      "GCF_000721105.1 removed. Contigs: 182 N50: 192242\n",
      "GCF_000721155.1 removed. Contigs: 158 N50: 174329\n",
      "GCF_000721175.1 removed. Contigs: 167 N50: 178406\n",
      "GCF_000721185.1 removed. Contigs: 139 N50: 201589\n",
      "GCF_000721235.1 removed. Contigs: 188 N50: 199385\n",
      "GCF_000721255.1 removed. Contigs: 270 N50: 129169\n",
      "GCF_000721265.1 removed. Contigs: 168 N50: 454671\n",
      "GCF_000721305.1 removed. Contigs: 257 N50: 108097\n",
      "GCF_000721335.1 removed. Contigs: 167 N50: 163325\n",
      "GCF_000721345.1 removed. Contigs: 368 N50: 66847\n",
      "GCF_000721365.1 removed. Contigs: 215 N50: 134293\n",
      "GCF_000721415.1 removed. Contigs: 152 N50: 107760\n",
      "GCF_000721435.1 removed. Contigs: 199 N50: 133692\n",
      "GCF_000721445.1 removed. Contigs: 167 N50: 323733\n",
      "GCF_000721455.1 removed. Contigs: 130 N50: 159310\n",
      "GCF_000721495.1 removed. Contigs: 250 N50: 107840\n",
      "GCF_000721515.1 removed. Contigs: 231 N50: 122082\n",
      "GCF_000721525.1 removed. Contigs: 260 N50: 160615\n",
      "GCF_000721535.1 removed. Contigs: 369 N50: 159554\n",
      "GCF_000721605.1 removed. Contigs: 179 N50: 296163\n",
      "GCF_000721665.1 removed. Contigs: 176 N50: 213895\n",
      "GCF_000721715.1 removed. Contigs: 171 N50: 257525\n",
      "GCF_000721935.1 removed. Contamination: 0.35 Completeness: 87.27\n",
      "GCF_000721945.1 removed. Contigs: 814 N50: 22397\n",
      "GCF_000721955.1 removed. Contigs: 106 N50: 169644\n",
      "GCF_000722185.1 removed. Contigs: 327 N50: 229674\n",
      "GCF_000725465.1 removed. Contigs: 225 N50: 232827\n",
      "GCF_000725475.1 removed. Contigs: 218 N50: 256603\n",
      "GCF_000725495.1 removed. Contigs: 131 N50: 317517\n",
      "GCF_000725545.1 removed. Contigs: 129 N50: 254413\n",
      "GCF_000725555.1 removed. Contigs: 127 N50: 254674\n",
      "GCF_000725565.1 removed. Contigs: 125 N50: 220172\n",
      "GCF_000725575.1 removed. Contigs: 124 N50: 184782\n",
      "GCF_000725625.1 removed. Contigs: 167 N50: 141967\n",
      "GCF_000725635.1 removed. Contigs: 123 N50: 243609\n",
      "GCF_000725645.1 removed. Contigs: 132 N50: 294936\n",
      "GCF_000725655.1 removed. Contigs: 939 N50: 225947\n",
      "GCF_000725715.1 removed. Contigs: 599 N50: 20270\n",
      "GCF_000725725.1 removed. Contigs: 147 N50: 246997\n",
      "GCF_000725745.1 removed. Contigs: 205 N50: 136678\n",
      "GCF_000725785.1 removed. Contigs: 141 N50: 285929\n",
      "GCF_000725805.1 removed. Contigs: 163 N50: 253485\n",
      "GCF_000725885.1 removed. Contigs: 104 N50: 245715\n",
      "GCF_000787755.1 removed. Contigs: 155 N50: 132365\n",
      "GCF_000787775.1 removed. Contigs: 139 N50: 125279\n",
      "GCF_000787815.1 removed. Contigs: 184 N50: 96689\n",
      "GCF_000787835.1 removed. Contigs: 107 N50: 144573\n",
      "GCF_000787855.1 removed. Contigs: 153 N50: 107739\n",
      "GCF_000797385.1 removed. Contigs: 295 N50: 68754\n",
      "GCF_000816485.1 removed. Contigs: 164 N50: 139282\n",
      "GCF_000955965.1 removed. Contigs: 233 N50: 220724\n",
      "GCF_000956015.1 removed. Contigs: 437 N50: 31162\n",
      "GCF_000958545.1 removed. Contigs: 115 N50: 386995\n",
      "GCF_000961885.1 removed. Contigs: 411 N50: 74283\n",
      "GCF_000966955.1 removed. Contigs: 1416 N50: 14614\n",
      "GCF_000966965.1 removed. Contigs: 434 N50: 45490\n",
      "GCF_000966975.1 removed. Contigs: 1874 N50: 8690\n",
      "GCF_000968685.2 removed. Contigs: 282 N50: 135927\n",
      "GCF_000972385.1 removed. Contigs: 856 N50: 91294\n",
      "GCF_000972395.1 removed. Contigs: 242 N50: 88608\n",
      "GCF_000972435.1 removed. Contigs: 548 N50: 28950\n",
      "GCF_000974985.2 removed. Contigs: 199 N50: 108867\n",
      "GCF_000978515.1 removed. Contigs: 989 N50: 14661\n",
      "GCF_000980885.2 removed. Contigs: 235 N50: 123175\n",
      "GCF_000981895.1 removed. Contigs: 648 N50: 23420\n",
      "GCF_000988945.1 removed. Contigs: 1056 N50: 16369\n",
      "GCF_001005085.2 removed. Contigs: 214 N50: 126586\n",
      "GCF_001005405.1 removed. Contigs: 1091 N50: 15103\n",
      "GCF_001008135.1 removed. Contigs: 909 N50: 19141\n",
      "GCF_001008345.1 removed. Contigs: 172 N50: 101336\n",
      "GCF_001014595.1 removed. Contigs: 122 N50: 120255\n",
      "GCF_001044185.1 removed. Contigs: 117 N50: 199354\n",
      "GCF_001044425.1 removed. Contigs: 163 N50: 95271\n",
      "GCF_001047315.1 removed. Contigs: 502 N50: 60064\n",
      "GCF_001047325.1 removed. Contigs: 329 N50: 120470\n",
      "GCF_001083795.1 removed. Contigs: 691 N50: 19693\n",
      "GCF_001182915.1 removed. Contigs: 1461 N50: 11328\n",
      "GCF_001188955.3 removed. Contigs: 107 N50: 228236\n",
      "GCF_001190165.1 removed. Contigs: 208 N50: 82381\n",
      "GCF_001270025.1 removed. Contigs: 350 N50: 54489\n",
      "GCF_001270485.1 removed. Contigs: 307 N50: 99945\n",
      "GCF_001270495.1 removed. Contigs: 277 N50: 113641\n",
      "GCF_001270505.1 removed. Contigs: 244 N50: 106664\n",
      "GCF_001270515.1 removed. Contigs: 424 N50: 62708\n",
      "GCF_001270565.1 removed. Contigs: 403 N50: 82152\n",
      "GCF_001270575.1 removed. Contigs: 304 N50: 75047\n",
      "GCF_001270585.1 removed. Contigs: 382 N50: 56727\n",
      "GCF_001270635.1 removed. Contigs: 343 N50: 81290\n",
      "GCF_001270675.1 removed. Contigs: 102 N50: 157297\n",
      "GCF_001279005.1 removed. Contigs: 288 N50: 158158\n",
      "GCF_001279015.1 removed. Contigs: 288 N50: 81262\n",
      "GCF_001279025.1 removed. Contigs: 310 N50: 89378\n",
      "GCF_001279065.1 removed. Contigs: 299 N50: 83235\n",
      "GCF_001279075.1 removed. Contigs: 336 N50: 79020\n",
      "GCF_001279095.1 removed. Contigs: 305 N50: 88462\n",
      "GCF_001279105.1 removed. Contigs: 365 N50: 70517\n",
      "GCF_001279345.1 removed. Contigs: 436 N50: 55924\n",
      "GCF_001279355.1 removed. Contigs: 291 N50: 92178\n",
      "GCF_001279365.1 removed. Contigs: 327 N50: 68059\n",
      "GCF_001279375.1 removed. Contigs: 367 N50: 80348\n",
      "GCF_001279425.1 removed. Contigs: 170 N50: 132393\n",
      "GCF_001279445.1 removed. Contigs: 218 N50: 94059\n",
      "GCF_001279455.1 removed. Contigs: 315 N50: 108182\n",
      "GCF_001279485.1 removed. Contigs: 315 N50: 78963\n",
      "GCF_001279495.1 removed. Contigs: 283 N50: 74316\n",
      "GCF_001279545.1 removed. Contigs: 264 N50: 164402\n",
      "GCF_001279605.1 removed. Contigs: 210 N50: 166107\n",
      "GCF_001279615.1 removed. Contigs: 226 N50: 140163\n",
      "GCF_001279625.1 removed. Contigs: 324 N50: 98566\n",
      "GCF_001279655.1 removed. Contigs: 259 N50: 91629\n",
      "GCF_001279685.1 removed. Contigs: 260 N50: 94956\n",
      "GCF_001279695.1 removed. Contigs: 362 N50: 90898\n",
      "GCF_001279725.1 removed. Contigs: 479 N50: 78201\n",
      "GCF_001279735.1 removed. Contigs: 190 N50: 111333\n",
      "GCF_001279765.1 removed. Contigs: 255 N50: 114603\n",
      "GCF_001279775.1 removed. Contigs: 294 N50: 98746\n",
      "GCF_001279805.1 removed. Contigs: 528 N50: 46679\n",
      "GCF_001279815.1 removed. Contigs: 202 N50: 130425\n",
      "GCF_001279845.1 removed. Contigs: 254 N50: 97762\n",
      "GCF_001279855.1 removed. Contigs: 232 N50: 103181\n",
      "GCF_001279885.1 removed. Contigs: 220 N50: 130619\n",
      "GCF_001279905.1 removed. Contigs: 260 N50: 104603\n",
      "GCF_001279925.1 removed. Contigs: 207 N50: 102927\n",
      "GCF_001279935.1 removed. Contigs: 258 N50: 86405\n",
      "GCF_001279965.1 removed. Contigs: 216 N50: 119376\n",
      "GCF_001279985.1 removed. Contigs: 425 N50: 126856\n",
      "GCF_001280005.1 removed. Contigs: 500 N50: 57397\n",
      "GCF_001280015.1 removed. Contigs: 273 N50: 150870\n",
      "GCF_001280045.1 removed. Contigs: 374 N50: 101769\n",
      "GCF_001280065.1 removed. Contigs: 184 N50: 209231\n",
      "GCF_001280095.1 removed. Contigs: 287 N50: 69817\n",
      "GCF_001280125.1 removed. Contigs: 287 N50: 69469\n",
      "GCF_001280885.1 removed. Contigs: 243 N50: 135294\n",
      "GCF_001282115.1 removed. Contigs: 692 N50: 22932\n",
      "GCF_001294335.1 removed. Contigs: 217 N50: 166063\n",
      "GCF_001294345.1 removed. Contigs: 238 N50: 322709\n",
      "GCF_001418125.1 removed. Contigs: 754 N50: 16752\n",
      "GCF_001418135.1 removed. Contigs: 707 N50: 17878\n",
      "GCF_001418165.1 removed. Contigs: 796 N50: 15219\n",
      "GCF_001418175.1 removed. Contigs: 811 N50: 16823\n",
      "GCF_001418325.1 removed. Contigs: 703 N50: 24134\n",
      "GCF_001418335.1 removed. Contigs: 561 N50: 29280\n",
      "GCF_001418385.1 removed. Contigs: 578 N50: 23290\n",
      "GCF_001418395.1 removed. Contigs: 1027 N50: 9198\n",
      "GCF_001418415.1 removed. Contigs: 783 N50: 18113\n",
      "GCF_001418465.1 removed. Contigs: 727 N50: 24820\n",
      "GCF_001418475.1 removed. Contigs: 1008 N50: 16332\n",
      "GCF_001418495.1 removed. Contigs: 743 N50: 11466\n",
      "GCF_001418505.1 removed. Contigs: 644 N50: 18934\n",
      "GCF_001418545.1 removed. Contigs: 940 N50: 11479\n",
      "GCF_001418565.1 removed. Contigs: 859 N50: 19214\n",
      "GCF_001418575.1 removed. Contigs: 802 N50: 17726\n",
      "GCF_001418585.1 removed. Contigs: 963 N50: 18170\n",
      "GCF_001418625.1 removed. Contigs: 767 N50: 18132\n",
      "GCF_001418645.1 removed. Contigs: 1411 N50: 10145\n",
      "GCF_001418655.1 removed. Contigs: 702 N50: 29593\n",
      "GCF_001419685.1 removed. Contigs: 631 N50: 25418\n",
      "GCF_001419695.1 removed. Contigs: 1058 N50: 8888\n",
      "GCF_001419705.1 removed. Contigs: 860 N50: 12922\n",
      "GCF_001419745.1 removed. Contigs: 732 N50: 20506\n",
      "GCF_001419765.1 removed. Contigs: 1055 N50: 15874\n",
      "GCF_001419795.1 removed. Contigs: 1456 N50: 10150\n",
      "GCF_001469455.2 removed. Contigs: 183 N50: 137942\n",
      "GCF_001484625.1 removed. Contigs: 225 N50: 61612\n",
      "GCF_001485105.1 removed. Contigs: 278 N50: 82196\n",
      "GCF_001485125.1 removed. Contigs: 147 N50: 193185\n",
      "GCF_001493375.1 removed. Contigs: 143 N50: 569072\n",
      "GCF_001507395.1 removed. Contigs: 315 N50: 87969\n",
      "GCF_001507435.1 removed. Contigs: 312 N50: 153060\n",
      "GCF_001509475.1 removed. Contigs: 424 N50: 73780\n",
      "GCF_001509485.1 removed. Contigs: 297 N50: 62597\n",
      "GCF_001509505.1 removed. Contigs: 282 N50: 104904\n",
      "GCF_001509565.1 removed. Contigs: 287 N50: 67140\n",
      "GCF_001509775.1 removed. Contigs: 439 N50: 140380\n",
      "GCF_001509785.1 removed. Contigs: 144 N50: 128237\n",
      "GCF_001509795.1 removed. Contigs: 137 N50: 134487\n",
      "GCF_001513965.1 removed. Contigs: 129 N50: 265228\n",
      "GCF_001514305.1 removed. Contigs: 152 N50: 214515\n",
      "GCF_001517125.1 removed. Contamination: 6.08 Completeness: 87.98\n",
      "GCF_001541145.1 removed. Contigs: 307 N50: 88399\n",
      "GCF_001550215.1 removed. Contigs: 1383 N50: 13060\n",
      "GCF_001550225.1 removed. Contigs: 1798 N50: 10646\n",
      "GCF_001550235.1 removed. Contamination: 6.48 Completeness: 95.29\n",
      "GCF_001550245.1 removed. Contigs: 1650 N50: 10930\n",
      "GCF_001550295.1 removed. Contigs: 1330 N50: 12509\n",
      "GCF_001550315.1 removed. Contigs: 1875 N50: 8672\n",
      "GCF_001550325.1 removed. Contigs: 1080 N50: 17738\n",
      "GCF_001550375.1 removed. Contigs: 931 N50: 21182\n",
      "GCF_001553435.1 removed. Contigs: 133 N50: 179494\n",
      "GCF_001553455.1 removed. Contigs: 680 N50: 28354\n",
      "GCF_001553475.1 removed. Contigs: 539 N50: 40212\n",
      "GCF_001570645.1 removed. Contigs: 217 N50: 97148\n",
      "GCF_001572115.1 removed. Contigs: 1464 N50: 12557\n",
      "GCF_001579685.1 removed. Contigs: 1173 N50: 9944\n",
      "GCF_001595515.1 removed. Contigs: 264 N50: 58042\n",
      "GCF_001642695.1 removed. Contigs: 197 N50: 4553814\n",
      "GCF_001642995.1 removed. Contigs: 207 N50: 86423\n",
      "GCF_001646665.1 removed. Contigs: 413 N50: 225771\n",
      "GCF_001652895.1 removed. Contigs: 259 N50: 62902\n",
      "GCF_001653115.1 removed. Contigs: 139 N50: 143984\n",
      "GCF_001653515.1 removed. Contamination: 3.24 Completeness: 87.67\n",
      "GCF_001661615.1 removed. Contigs: 902 N50: 14733\n",
      "GCF_001700505.1 removed. Contigs: 410 N50: 35885\n",
      "GCF_001700515.1 removed. Contigs: 732 N50: 21630\n",
      "GCF_001704635.1 removed. Contigs: 276 N50: 72375\n",
      "GCF_001705785.1 removed. Contigs: 125 N50: 317920\n",
      "GCF_001723055.1 removed. Contigs: 608 N50: 24673\n",
      "GCF_001723065.1 removed. Contigs: 1885 N50: 7160\n",
      "GCF_001723075.1 removed. Contigs: 1229 N50: 11948\n",
      "GCF_001723115.1 removed. Contigs: 1108 N50: 13772\n",
      "GCF_001746285.1 removed. Contigs: 236 N50: 67345\n",
      "GCF_001746305.1 removed. Contigs: 131 N50: 147588\n",
      "GCF_001746315.1 removed. Contigs: 162 N50: 100704\n",
      "GCF_001746365.1 removed. Contigs: 132 N50: 137390\n",
      "GCF_001748305.1 removed. Contigs: 262 N50: 56991\n",
      "GCF_001751245.1 removed. Contigs: 166 N50: 132276\n",
      "GCF_001751255.1 removed. Contigs: 731 N50: 19859\n",
      "GCF_001751305.1 removed. Contamination: 6.91 Completeness: 97.99\n",
      "GCF_001756925.1 removed. Contigs: 320 N50: 51843\n",
      "GCF_001865245.1 removed. Contigs: 240 N50: 49045\n",
      "GCF_001865255.1 removed. Contigs: 876 N50: 12619\n",
      "GCF_001866645.1 removed. Contigs: 166 N50: 99963\n",
      "GCF_001866665.1 removed. Contigs: 218 N50: 159229\n",
      "GCF_001866675.1 removed. Contigs: 174 N50: 162032\n",
      "GCF_001879105.1 removed. Contigs: 206 N50: 45324\n",
      "GCF_001906585.1 removed. Contigs: 170 N50: 110055\n",
      "GCF_001941265.1 removed. Contigs: 1211 N50: 16535\n",
      "GCF_001941275.1 removed. Contigs: 983 N50: 20309\n",
      "GCF_001941285.1 removed. Contigs: 1073 N50: 20017\n",
      "GCF_001941325.1 removed. Contigs: 993 N50: 19323\n",
      "GCF_001949645.1 removed. Contigs: 259 N50: 85309\n",
      "GCF_001953875.1 removed. Contigs: 208 N50: 63430\n",
      "GCF_001969965.1 removed. Contigs: 646 N50: 28947\n",
      "GCF_002005565.1 removed. Contigs: 170 N50: 99308\n",
      "GCF_002007125.1 removed. Contigs: 127 N50: 183690\n",
      "GCF_002078235.1 removed. Contigs: 986 N50: 15779\n",
      "GCF_002114215.1 removed. Contigs: 1513 N50: 10426\n",
      "GCF_002119195.1 removed. Contigs: 126 N50: 114666\n",
      "GCF_002119225.1 removed. Contigs: 352 N50: 30888\n",
      "GCF_002148965.1 removed. Contigs: 870 N50: 21344\n",
      "GCF_002150735.1 removed. Contigs: 736 N50: 27371\n",
      "GCF_002150845.1 removed. Contigs: 852 N50: 18609\n",
      "GCF_002154275.1 removed. Contigs: 1258 N50: 17428\n",
      "GCF_002154345.1 removed. Contigs: 846 N50: 17809\n",
      "GCF_002154375.1 removed. Contigs: 1135 N50: 13850\n",
      "GCF_002154385.1 removed. Contigs: 594 N50: 23195\n",
      "GCF_002154415.1 removed. Contigs: 874 N50: 14867\n",
      "GCF_002154445.1 removed. Contigs: 1555 N50: 6532\n",
      "GCF_002154495.1 removed. Contigs: 981 N50: 16615\n",
      "GCF_002154505.1 removed. Contigs: 824 N50: 27358\n",
      "GCF_002154555.1 removed. Contigs: 1088 N50: 14213\n",
      "GCF_002154575.1 removed. Contigs: 946 N50: 17336\n",
      "GCF_002154605.1 removed. Contigs: 1328 N50: 17761\n",
      "GCF_002154615.1 removed. Contigs: 926 N50: 12194\n",
      "GCF_002154675.1 removed. Contigs: 897 N50: 22762\n",
      "GCF_002155725.1 removed. Contigs: 1401 N50: 14164\n",
      "GCF_002155735.1 removed. Contigs: 1748 N50: 10223\n",
      "GCF_002155765.1 removed. Contigs: 1566 N50: 11589\n",
      "GCF_002155885.1 removed. Contigs: 1370 N50: 17562\n",
      "GCF_002155895.1 removed. Contigs: 808 N50: 20398\n",
      "GCF_002155905.1 removed. Contigs: 1107 N50: 15457\n",
      "GCF_002155915.1 removed. Contigs: 595 N50: 21010\n",
      "GCF_002156055.1 removed. Contigs: 757 N50: 18965\n",
      "GCF_002156075.1 removed. Contigs: 1568 N50: 12428\n",
      "GCF_002156095.1 removed. Contigs: 692 N50: 27328\n",
      "GCF_002156145.1 removed. Contigs: 1243 N50: 12317\n",
      "GCF_002158965.1 removed. Contigs: 650 N50: 22501\n",
      "GCF_002224125.1 removed. Contigs: 127 N50: 254229\n",
      "GCF_002242735.1 removed. Contigs: 285 N50: 78584\n",
      "GCF_002242805.1 removed. Contigs: 242 N50: 262623\n",
      "GCF_002276245.1 removed. Contigs: 581 N50: 26110\n",
      "GCF_002277855.1 removed. Contigs: 679 N50: 34813\n",
      "GCF_002286695.1 removed. Contigs: 714 N50: 21466\n",
      "GCF_002289305.1 removed. Contigs: 1153 N50: 8176\n",
      "GCF_002289335.1 removed. Contigs: 1053 N50: 10498\n",
      "GCF_002291145.1 removed. Contigs: 222 N50: 103532\n",
      "GCF_002317005.1 removed. Contigs: 1114 N50: 9968\n",
      "GCF_002317025.1 removed. Contigs: 1028 N50: 11431\n",
      "GCF_002317035.1 removed. Contigs: 957 N50: 12358\n",
      "GCF_002335465.1 removed. Contigs: 131 N50: 304660\n",
      "GCF_002382885.1 removed. Contigs: 189 N50: 85451\n",
      "GCF_002551015.1 removed. Contigs: 1400 N50: 11691\n",
      "GCF_002551045.1 removed. Contigs: 1178 N50: 15426\n",
      "GCF_002551055.1 removed. Contigs: 892 N50: 20004\n",
      "GCF_002551065.1 removed. Contigs: 337 N50: 42533\n",
      "GCF_002551105.1 removed. Contigs: 555 N50: 29441\n",
      "GCF_002551135.1 removed. Contigs: 504 N50: 31001\n",
      "GCF_002551145.1 removed. Contigs: 559 N50: 29129\n",
      "GCF_002551175.1 removed. Contigs: 214 N50: 95964\n",
      "GCF_002551185.1 removed. Contigs: 494 N50: 32985\n",
      "GCF_002551215.1 removed. Contigs: 784 N50: 21085\n",
      "GCF_002551225.1 removed. Contigs: 670 N50: 24278\n",
      "GCF_002551235.1 removed. Contigs: 619 N50: 25776\n",
      "GCF_002551245.1 removed. Contigs: 629 N50: 26146\n",
      "GCF_002551295.1 removed. Contigs: 228 N50: 70290\n",
      "GCF_002551305.1 removed. Contigs: 245 N50: 81082\n",
      "GCF_002551335.1 removed. Contigs: 918 N50: 15476\n",
      "GCF_002551345.1 removed. Contigs: 164 N50: 83606\n",
      "GCF_002551375.1 removed. Contigs: 164 N50: 103747\n",
      "GCF_002551415.1 removed. Contigs: 619 N50: 23604\n",
      "GCF_002551425.1 removed. Contigs: 1410 N50: 14459\n",
      "GCF_002551435.1 removed. Contigs: 145 N50: 109544\n",
      "GCF_002551465.1 removed. Contigs: 222 N50: 80138\n",
      "GCF_002551475.1 removed. Contigs: 222 N50: 63977\n",
      "GCF_002573955.1 removed. Contigs: 663 N50: 20124\n",
      "GCF_002705665.1 removed. Contigs: 106 N50: 261762\n",
      "GCF_002727115.1 removed. Contamination: 7.93 Completeness: 99.64\n",
      "GCF_002843305.1 removed. Contigs: 119 N50: 127012\n",
      "GCF_002844215.1 removed. Contigs: 182 N50: 84238\n",
      "GCF_002891435.1 removed. Contigs: 1024 N50: 17129\n",
      "GCF_002891865.1 removed. Contamination: 5.26 Completeness: 100.0\n",
      "GCF_002907485.1 removed. Contigs: 111 N50: 148637\n",
      "GCF_002910755.1 removed. Contigs: 639 N50: 18032\n",
      "GCF_002910795.1 removed. Contigs: 1592 N50: 7255\n",
      "GCF_002910815.1 removed. Contigs: 1369 N50: 7818\n",
      "GCF_002910825.1 removed. Contigs: 1057 N50: 13915\n",
      "GCF_002910855.1 removed. Contigs: 910 N50: 12271\n",
      "GCF_002910875.1 removed. Contigs: 388 N50: 41121\n",
      "GCF_002910895.1 removed. Contigs: 496 N50: 30992\n",
      "GCF_002910905.1 removed. Contigs: 311 N50: 45758\n",
      "GCF_002910915.1 removed. Contigs: 195 N50: 76497\n",
      "GCF_002910935.1 removed. Contigs: 580 N50: 26253\n",
      "GCF_002910985.1 removed. Contigs: 609 N50: 24967\n",
      "GCF_002911015.1 removed. Contigs: 277 N50: 75083\n",
      "GCF_002920535.1 removed. Contigs: 193 N50: 85687\n",
      "GCF_002920575.1 removed. Contigs: 245 N50: 65242\n",
      "GCF_002920615.1 removed. Contigs: 116 N50: 235045\n",
      "GCF_002920635.1 removed. Contigs: 279 N50: 60931\n",
      "GCF_002939385.1 removed. Contigs: 190 N50: 81228\n",
      "GCF_002953255.1 removed. Contamination: 0.79 Completeness: 86.61\n",
      "GCF_002954775.1 removed. Contamination: 0.79 Completeness: 88.84\n",
      "GCF_002966105.1 removed. Contigs: 150 N50: 424085\n",
      "GCF_002982015.1 removed. Contigs: 748 N50: 13225\n",
      "GCF_003013745.1 removed. Contigs: 236 N50: 70608\n",
      "GCF_003073355.1 removed. Contigs: 396 N50: 38992\n",
      "GCF_003122345.2 removed. Contigs: 150 N50: 459751\n",
      "GCF_003129505.1 removed. Contigs: 281 N50: 83624\n",
      "GCF_003143855.1 removed. Contigs: 169 N50: 160697\n",
      "GCF_003173635.1 removed. Contigs: 1956 N50: 6332\n",
      "GCF_003173645.1 removed. Contigs: 1836 N50: 7225\n",
      "GCF_003205575.1 removed. Contigs: 439 N50: 76804\n",
      "GCF_003236365.1 removed. Contigs: 1200 N50: 12812\n",
      "GCF_003258295.1 removed. Contigs: 281 N50: 56286\n",
      "GCF_003259365.1 removed. Contigs: 111 N50: 362350\n",
      "GCF_003268555.1 removed. Contigs: 1013 N50: 17536\n",
      "GCF_003270085.1 removed. Contigs: 298 N50: 33934\n",
      "GCF_003311645.1 removed. Contigs: 296 N50: 121713\n",
      "GCF_003344905.1 removed. Contigs: 118 N50: 229990\n",
      "GCF_003346515.1 removed. Contigs: 480 N50: 28347\n",
      "GCF_003389455.1 removed. Contigs: 486 N50: 33076\n",
      "GCF_003429565.1 removed. Contigs: 239 N50: 58844\n",
      "GCF_003501885.1 removed. Contigs: 466 N50: 30911\n",
      "GCF_003568675.1 removed. Contigs: 352 N50: 59981\n",
      "GCF_003584805.1 removed. Contigs: 1015 N50: 18661\n",
      "GCF_003584815.1 removed. Contigs: 1211 N50: 13310\n",
      "GCF_003584825.1 removed. Contigs: 1078 N50: 17518\n",
      "GCF_003626645.1 removed. Contigs: 247 N50: 541812\n",
      "GCF_003671675.1 removed. Contigs: 648 N50: 18398\n",
      "GCF_003671685.1 removed. Contigs: 617 N50: 23688\n",
      "GCF_003671755.1 removed. Contigs: 833 N50: 15941\n",
      "GCF_003688995.1 removed. Contigs: 137 N50: 212672\n",
      "GCF_003696235.1 removed. Contigs: 428 N50: 25828\n",
      "GCF_003721215.1 removed. Contigs: 886 N50: 24675\n",
      "GCF_003851625.1 removed. Contigs: 292 N50: 103610\n",
      "GCF_003933485.1 removed. Contigs: 131 N50: 460998\n",
      "GCF_003933965.1 removed. Contigs: 1347 N50: 13705\n",
      "GCF_003947455.1 removed. Contigs: 465 N50: 37062\n",
      "GCF_003947555.1 removed. Contigs: 790 N50: 23520\n",
      "GCF_003947585.1 removed. Contigs: 975 N50: 21249\n",
      "GCF_003949505.1 removed. Contigs: 296 N50: 54438\n",
      "GCF_003949535.1 removed. Contigs: 1104 N50: 13955\n",
      "GCF_003949665.1 removed. Contigs: 963 N50: 17391\n",
      "GCF_003949695.1 removed. Contigs: 880 N50: 22710\n",
      "GCF_003949795.1 removed. Contigs: 821 N50: 20356\n",
      "GCF_003949805.1 removed. Contigs: 753 N50: 19785\n",
      "GCF_003949835.1 removed. Contamination: 6.18 Completeness: 100.0\n",
      "GCF_003949875.1 removed. Contigs: 118 N50: 177740\n",
      "GCF_003949895.1 removed. Contigs: 1916 N50: 8678\n",
      "GCF_003949955.1 removed. Contigs: 926 N50: 14397\n",
      "GCF_003977835.1 removed. Contigs: 214 N50: 141168\n",
      "GCF_003977875.1 removed. Contigs: 619 N50: 37166\n",
      "GCF_003984785.1 removed. Contigs: 619 N50: 25940\n",
      "GCF_003999195.1 removed. Contigs: 412 N50: 40353\n",
      "GCF_004023625.1 removed. Contigs: 131 N50: 275254\n",
      "GCF_004078575.1 removed. Contigs: 1057 N50: 11575\n",
      "GCF_004117935.1 removed. Contigs: 181 N50: 83322\n",
      "GCF_004122735.1 removed. Contigs: 289 N50: 54690\n",
      "GCF_004122755.1 removed. Contigs: 277 N50: 33909\n",
      "GCF_004151105.1 removed. Contigs: 172 N50: 203880\n",
      "GCF_004187715.1 removed. Contigs: 171 N50: 601559\n",
      "GCF_004196055.1 removed. Contigs: 146 N50: 113214\n",
      "GCF_004196115.1 removed. Contigs: 147 N50: 97051\n",
      "GCF_004196335.1 removed. Contigs: 333 N50: 314273\n",
      "GCF_004212445.1 removed. Contigs: 235 N50: 90629\n",
      "GCF_004305975.1 removed. Contigs: 752 N50: 18370\n",
      "GCF_004320085.1 removed. Contigs: 962 N50: 15111\n",
      "GCF_004348315.1 removed. Contigs: 703 N50: 17808\n",
      "GCF_004348415.1 removed. Contigs: 951 N50: 13702\n",
      "GCF_004403015.1 removed. Contigs: 175 N50: 87699\n",
      "GCF_004684775.1 removed. Contigs: 630 N50: 25361\n",
      "GCF_004684805.1 removed. Contigs: 801 N50: 16138\n",
      "GCF_004768505.1 removed. Contigs: 120 N50: 489196\n",
      "GCF_004794175.1 removed. Contigs: 807 N50: 15942\n",
      "GCF_004794975.1 removed. Contigs: 110 N50: 210358\n",
      "GCF_004795015.1 removed. Contigs: 115 N50: 114260\n",
      "GCF_004795035.1 removed. Contigs: 113 N50: 166249\n",
      "GCF_004795055.1 removed. Contigs: 412 N50: 166286\n",
      "GCF_004795115.1 removed. Contigs: 106 N50: 235884\n",
      "GCF_004795155.1 removed. Contigs: 219 N50: 72620\n",
      "GCF_005047355.1 removed. Contigs: 389 N50: 131738\n",
      "GCF_005280195.1 removed. Contigs: 111 N50: 326353\n",
      "GCF_005768555.2 removed. Contigs: 352 N50: 59982\n",
      "GCF_005786655.1 removed. Contigs: 168 N50: 137451\n",
      "GCF_005981925.1 removed. Contigs: 795 N50: 18694\n",
      "GCF_005981935.1 removed. Contigs: 1451 N50: 9479\n",
      "GCF_005981965.1 removed. Contigs: 823 N50: 17998\n",
      "GCF_006335115.1 removed. Contigs: 186 N50: 117311\n",
      "GCF_006539165.1 removed. Contigs: 197 N50: 142848\n",
      "GCF_006539185.1 removed. Contigs: 147 N50: 146657\n",
      "GCF_006539505.1 removed. Contigs: 218 N50: 109659\n",
      "GCF_006547165.1 removed. Contigs: 178 N50: 144065\n",
      "GCF_006547175.1 removed. Contigs: 345 N50: 60995\n",
      "GCF_006547185.1 removed. Contigs: 377 N50: 58885\n",
      "GCF_007050945.1 removed. Contigs: 109 N50: 375098\n",
      "GCF_007096885.1 removed. Contigs: 143 N50: 141317\n",
      "GCF_007097205.2 removed. Contigs: 769 N50: 15903\n",
      "GCF_007097665.2 removed. Contigs: 1861 N50: 5488\n",
      "GCF_007280575.1 removed. Contigs: 814 N50: 16345\n",
      "GCF_007280585.1 removed. Contigs: 1118 N50: 11372\n",
      "GCF_007677905.1 removed. Contamination: 12.970456661443787 Completeness: 100.0\n",
      "GCF_007679265.1 removed. Contigs: 138 N50: 224298\n",
      "GCF_007679425.1 removed. Contigs: 141 N50: 221527\n",
      "GCF_007896895.1 removed. Contigs: 210 N50: 70083\n",
      "GCF_008039505.1 removed. Contigs: 188 N50: 220695\n",
      "GCF_008042045.1 removed. Contigs: 292 N50: 1120392\n",
      "GCF_008042075.1 removed. Contigs: 190 N50: 1367314\n",
      "GCF_008042115.1 removed. Contigs: 301 N50: 493345\n",
      "GCF_008042125.1 removed. Contigs: 261 N50: 388359\n",
      "GCF_008042145.1 removed. Contigs: 346 N50: 468649\n",
      "GCF_008042155.1 removed. Contigs: 190 N50: 1700478\n",
      "GCF_008042275.1 removed. Contigs: 140 N50: 4652464\n",
      "GCF_008086975.1 removed. Contigs: 882 N50: 16068\n",
      "GCF_008087005.1 removed. Contigs: 930 N50: 20394\n",
      "GCF_008120935.1 removed. Contigs: 309 N50: 307497\n",
      "GCF_008180165.1 removed. Contigs: 729 N50: 28163\n",
      "GCF_008271805.1 removed. Contigs: 174 N50: 110600\n",
      "GCF_008369065.1 removed. Contigs: 309 N50: 296590\n",
      "GCF_008386495.1 removed. Contigs: 183 N50: 115964\n",
      "GCF_008632535.1 removed. Contigs: 546 N50: 28351\n",
      "GCF_008806595.1 removed. Contigs: 105 N50: 198006\n",
      "GCF_008868685.1 removed. Contigs: 127 N50: 170991\n",
      "GCF_008905045.1 removed. Contigs: 200 N50: 138999\n",
      "GCF_008932075.1 removed. Contigs: 121 N50: 161029\n",
      "GCF_009208535.1 removed. Contigs: 169 N50: 164101\n",
      "GCF_009377175.1 removed. Contigs: 563 N50: 32765\n",
      "GCF_009377185.1 removed. Contigs: 1115 N50: 15786\n",
      "GCF_009377205.1 removed. Contigs: 939 N50: 17201\n",
      "GCF_009377235.1 removed. Contigs: 544 N50: 34036\n",
      "GCF_009600885.1 removed. Contigs: 209 N50: 91474\n",
      "GCF_009600895.1 removed. Contigs: 412 N50: 37935\n",
      "GCF_009604385.1 removed. Contigs: 106 N50: 161565\n",
      "GCF_009709555.1 removed. Contigs: 253 N50: 97082\n",
      "GCF_009735685.1 removed. Contigs: 431 N50: 32682\n",
      "GCF_009739225.1 removed. Contigs: 618 N50: 22652\n",
      "GCF_009739465.1 removed. Contamination: 12.07 Completeness: 100.0\n",
      "GCF_009862025.1 removed. Contigs: 472 N50: 35596\n",
      "GCF_009862125.1 removed. Contamination: 4.72 Completeness: 87.98\n",
      "GCF_009862225.1 removed. Contigs: 120 N50: 169228\n",
      "GCF_009862265.1 removed. Contigs: 115 N50: 141016\n",
      "GCF_009862305.1 removed. Contigs: 111 N50: 362350\n",
      "GCF_009862355.1 removed. Contamination: 6.1 Completeness: 85.85\n",
      "GCF_009862405.1 removed. Contigs: 219 N50: 55778\n",
      "GCF_009862465.1 removed. Contigs: 351 N50: 38665\n",
      "GCF_009862475.1 removed. Contigs: 266 N50: 54854\n",
      "GCF_009862505.1 removed. Contigs: 193 N50: 181994\n",
      "GCF_009862515.1 removed. Contigs: 1579 N50: 9144\n",
      "GCF_009862625.1 removed. Contigs: 731 N50: 23142\n",
      "GCF_009862635.1 removed. Contamination: 8.75 Completeness: 93.67\n",
      "GCF_009862685.1 removed. Contamination: 6.54 Completeness: 98.25\n",
      "GCF_009862765.1 removed. Contamination: 17.67 Completeness: 94.54\n",
      "GCF_009862825.1 removed. Contigs: 959 N50: 13297\n",
      "GCF_009862845.1 removed. Contamination: 7.31 Completeness: 92.54\n",
      "GCF_009862935.1 removed. Contigs: 453 N50: 40361\n",
      "GCF_009863025.1 removed. Contigs: 731 N50: 18077\n",
      "GCF_009863035.1 removed. Contigs: 1065 N50: 13238\n",
      "GCF_009863045.1 removed. Contigs: 762 N50: 21477\n",
      "GCF_009863085.1 removed. Contigs: 106 N50: 261762\n",
      "GCF_009863095.1 removed. Contigs: 460 N50: 38736\n",
      "GCF_009863125.1 removed. Contigs: 142 N50: 266092\n",
      "GCF_009863145.1 removed. Contigs: 111 N50: 142272\n",
      "GCF_009863165.1 removed. Contigs: 128 N50: 129868\n",
      "GCF_009863175.1 removed. Contigs: 174 N50: 88536\n",
      "GCF_009863225.1 removed. Contigs: 254 N50: 59002\n",
      "GCF_009864805.1 removed. Contamination: 5.79 Completeness: 99.47\n",
      "GCF_009864855.1 removed. Contigs: 1923 N50: 5700\n",
      "GCF_009864985.1 removed. Contigs: 811 N50: 17565\n",
      "GCF_009865085.1 removed. Contigs: 111 N50: 113608\n",
      "GCF_009865165.1 removed. Contigs: 837 N50: 21184\n",
      "GCF_009865175.1 removed. Contigs: 993 N50: 17668\n",
      "GCF_009865205.1 removed. Contigs: 573 N50: 22006\n",
      "GCF_009865215.1 removed. Contigs: 1646 N50: 14460\n",
      "GCF_009865265.1 removed. Contigs: 1765 N50: 9124\n",
      "GCF_009865275.1 removed. Contigs: 397 N50: 35205\n",
      "GCF_009865385.1 removed. Contigs: 809 N50: 21568\n",
      "GCF_009865405.1 removed. Contigs: 1270 N50: 13092\n",
      "GCF_009865425.1 removed. Contigs: 777 N50: 17651\n",
      "GCF_009865475.1 removed. Contigs: 462 N50: 44392\n",
      "GCF_009865505.1 removed. Contigs: 1872 N50: 7364\n",
      "GCF_009865545.1 removed. Contigs: 1059 N50: 14193\n",
      "GCF_009865565.1 removed. Contigs: 500 N50: 36788\n",
      "GCF_009865585.1 removed. Contigs: 1902 N50: 8560\n",
      "GCF_009865595.1 removed. Contamination: 7.48 Completeness: 96.57\n",
      "GCF_009865605.1 removed. Contamination: 6.57 Completeness: 96.73\n",
      "GCF_009866455.1 removed. Contigs: 165 N50: 153923\n",
      "GCF_009870065.1 removed. Contigs: 1157 N50: 10784\n",
      "GCF_009870095.1 removed. Contigs: 620 N50: 27471\n",
      "GCF_009872515.1 removed. Contigs: 1059 N50: 24858\n",
      "GCF_009872535.1 removed. Contigs: 1124 N50: 15738\n",
      "GCF_009872555.1 removed. Contigs: 369 N50: 50997\n",
      "GCF_009872565.1 removed. Contigs: 560 N50: 25468\n",
      "GCF_009873275.1 removed. Contamination: 5.15 Completeness: 89.99\n",
      "GCF_009908195.1 removed. Contigs: 794 N50: 16908\n",
      "GCF_009916105.1 removed. Contigs: 854 N50: 17537\n",
      "GCF_010119355.1 removed. Contigs: 170 N50: 87030\n",
      "GCF_010548045.1 removed. Contigs: 426 N50: 37429\n",
      "GCF_010548075.1 removed. Contigs: 1631 N50: 11324\n",
      "GCF_010548115.1 removed. Contigs: 812 N50: 30742\n",
      "GCF_010548145.1 removed. Contigs: 521 N50: 25717\n",
      "GCF_010548175.1 removed. Contigs: 756 N50: 22224\n",
      "GCF_010548185.1 removed. Contigs: 1740 N50: 9141\n",
      "GCF_010548215.1 removed. Contigs: 1869 N50: 9238\n",
      "GCF_010548245.1 removed. Contigs: 1202 N50: 14469\n",
      "GCF_010548275.1 removed. Contigs: 183 N50: 217140\n",
      "GCF_010548305.1 removed. Contigs: 134 N50: 336850\n",
      "GCF_010548325.1 removed. Contigs: 268 N50: 163002\n",
      "GCF_010548345.1 removed. Contigs: 325 N50: 152276\n",
      "GCF_010548375.1 removed. Contigs: 923 N50: 394215\n",
      "GCF_010548385.1 removed. Contigs: 1070 N50: 16758\n",
      "GCF_010548435.1 removed. Contigs: 981 N50: 18965\n",
      "GCF_010548465.1 removed. Contigs: 908 N50: 16387\n",
      "GCF_010548475.1 removed. Contamination: 5.53 Completeness: 99.14\n",
      "GCF_010548505.1 removed. Contigs: 150 N50: 124267\n",
      "GCF_010548515.1 removed. Contigs: 104 N50: 184012\n",
      "GCF_010548525.1 removed. Contigs: 1499 N50: 13264\n",
      "GCF_010548565.1 removed. Contigs: 1034 N50: 16154\n",
      "GCF_010548575.1 removed. Contigs: 551 N50: 34865\n",
      "GCF_010548605.1 removed. Contigs: 583 N50: 28537\n",
      "GCF_010548615.1 removed. Contigs: 1117 N50: 15550\n",
      "GCF_010548665.1 removed. Contigs: 1956 N50: 8187\n",
      "GCF_010548675.1 removed. Contigs: 371 N50: 42839\n",
      "GCF_010548705.1 removed. Contigs: 415 N50: 41768\n",
      "GCF_010548715.1 removed. Contigs: 644 N50: 24401\n",
      "GCF_010548725.1 removed. Contigs: 652 N50: 20748\n",
      "GCF_010550245.1 removed. Contamination: 5.65 Completeness: 100.0\n",
      "GCF_010550325.1 removed. Contigs: 1317 N50: 14483\n",
      "GCF_010550365.1 removed. Contigs: 1799 N50: 7286\n",
      "GCF_010550375.1 removed. Contigs: 635 N50: 26576\n",
      "GCF_010550405.1 removed. Contigs: 1642 N50: 10146\n",
      "GCF_010550435.1 removed. Contamination: 8.62 Completeness: 98.28\n",
      "GCF_010550505.1 removed. Contigs: 1226 N50: 13806\n",
      "GCF_010550525.1 removed. Contigs: 1440 N50: 12106\n",
      "GCF_010550535.1 removed. Contigs: 672 N50: 23209\n",
      "GCF_010550605.1 removed. Contigs: 537 N50: 31997\n",
      "GCF_010550625.1 removed. Contigs: 613 N50: 22756\n",
      "GCF_010550645.1 removed. Contigs: 1631 N50: 8492\n",
      "GCF_010892295.1 removed. Contigs: 174 N50: 116762\n",
      "GCF_011008945.1 removed. Contigs: 1096 N50: 19372\n",
      "GCF_011044975.1 removed. Contigs: 613 N50: 28409\n",
      "GCF_011044995.1 removed. Contigs: 505 N50: 31168\n",
      "GCF_011045015.1 removed. Contigs: 496 N50: 41950\n",
      "GCF_011045025.1 removed. Contigs: 532 N50: 39427\n",
      "GCF_011045075.1 removed. Contigs: 1141 N50: 12556\n",
      "GCF_011055525.1 removed. Contigs: 143 N50: 110457\n",
      "GCF_011170145.1 removed. Contamination: 7.5 Completeness: 99.89\n",
      "GCF_011421595.1 removed. Contigs: 424 N50: 28352\n",
      "GCF_011766325.1 removed. Contigs: 120 N50: 264498\n",
      "GCF_012033735.1 removed. Contigs: 482 N50: 20985\n",
      "GCF_012033785.1 removed. Contigs: 547 N50: 17710\n",
      "GCF_012034175.1 removed. Contigs: 101 N50: 216213\n",
      "GCF_012034295.1 removed. Contigs: 914 N50: 15297\n",
      "GCF_012034305.1 removed. Contigs: 1252 N50: 12619\n",
      "GCF_012034385.1 removed. Contigs: 474 N50: 20130\n",
      "GCF_012328665.1 removed. Contigs: 791 N50: 15728\n",
      "GCF_013004455.1 removed. Contigs: 512 N50: 29851\n",
      "GCF_013018355.1 removed. Contigs: 382 N50: 131299\n",
      "GCF_013030945.1 removed. Contigs: 238 N50: 127665\n",
      "GCF_013044505.1 removed. Contigs: 569 N50: 40558\n",
      "GCF_013055795.1 removed. Contigs: 105 N50: 166849\n",
      "GCF_013177615.1 removed. Contigs: 211 N50: 147764\n",
      "GCF_013302945.1 removed. Contamination: 0.0 Completeness: 8.620689655172415\n",
      "GCF_013345665.1 removed. Contamination: 6.32 Completeness: 100.0\n",
      "GCF_013357925.1 removed. Contigs: 240 N50: 543791\n",
      "GCF_013357935.1 removed. Contigs: 333 N50: 44555\n",
      "GCF_013357965.1 removed. Contigs: 570 N50: 29740\n",
      "GCF_013358025.1 removed. Contigs: 480 N50: 32071\n",
      "GCF_013358045.1 removed. Contigs: 623 N50: 29740\n",
      "GCF_013358055.1 removed. Contigs: 904 N50: 29164\n",
      "GCF_013358065.1 removed. Contigs: 515 N50: 28882\n",
      "GCF_013358105.1 removed. Contigs: 502 N50: 29830\n",
      "GCF_013358115.1 removed. Contigs: 453 N50: 34436\n",
      "GCF_013358135.1 removed. Contigs: 493 N50: 30076\n",
      "GCF_013358155.1 removed. Contigs: 508 N50: 29219\n",
      "GCF_013358165.1 removed. Contigs: 614 N50: 26021\n",
      "GCF_013358235.1 removed. Contigs: 563 N50: 25668\n",
      "GCF_013358245.1 removed. Contigs: 516 N50: 29143\n",
      "GCF_013358305.1 removed. Contigs: 549 N50: 24483\n",
      "GCF_013358325.1 removed. Contigs: 523 N50: 29656\n",
      "GCF_013358335.1 removed. Contigs: 520 N50: 31126\n",
      "GCF_013362995.1 removed. Contigs: 469 N50: 26924\n",
      "GCF_013363485.1 removed. Contigs: 658 N50: 36637\n",
      "GCF_013363585.1 removed. Contigs: 447 N50: 44999\n",
      "GCF_013363615.1 removed. Contigs: 280 N50: 148248\n",
      "GCF_013363655.1 removed. Contigs: 258 N50: 108202\n",
      "GCF_013363675.1 removed. Contigs: 186 N50: 118325\n",
      "GCF_013363715.1 removed. Contigs: 330 N50: 65929\n",
      "GCF_013371965.1 removed. Contigs: 187 N50: 107473\n",
      "GCF_013387495.1 removed. Contigs: 160 N50: 191302\n",
      "GCF_013401435.1 removed. Contigs: 723 N50: 22797\n",
      "GCF_013421635.1 removed. Contigs: 1475 N50: 6270\n",
      "GCF_013433285.1 removed. Contigs: 168 N50: 147083\n",
      "GCF_013618515.1 removed. Contigs: 748 N50: 15600\n",
      "GCF_013618545.1 removed. Contigs: 473 N50: 26709\n",
      "GCF_013778455.1 removed. Contigs: 109 N50: 260785\n",
      "GCF_013870495.1 removed. Contigs: 1608 N50: 10913\n",
      "GCF_013870515.1 removed. Contigs: 497 N50: 20451\n",
      "GCF_013870535.1 removed. Contamination: 0.79 Completeness: 87.81\n",
      "GCF_013870595.1 removed. Contigs: 1416 N50: 7656\n",
      "GCF_013912435.1 removed. Contigs: 231 N50: 144039\n",
      "GCF_014141525.1 removed. Contigs: 1643 N50: 6206\n",
      "GCF_014141535.1 removed. Contigs: 1801 N50: 5021\n",
      "GCF_014156695.1 removed. Contigs: 1002 N50: 10866\n",
      "GCF_014257025.1 removed. Contigs: 210 N50: 119198\n",
      "GCF_014284355.1 removed. Contigs: 936 N50: 15211\n",
      "GCF_014284425.1 removed. Contigs: 1174 N50: 13902\n",
      "GCF_014493765.1 removed. Contigs: 178 N50: 363618\n",
      "GCF_014495685.1 removed. Contigs: 284 N50: 80030\n",
      "GCF_014534645.1 removed. Contigs: 241 N50: 166916\n",
      "GCF_014596445.1 removed. Contigs: 277 N50: 67094\n",
      "GCF_014596465.1 removed. Contigs: 642 N50: 32587\n",
      "GCF_014596505.1 removed. Contigs: 162 N50: 78797\n",
      "GCF_014596515.1 removed. Contamination: 7.11 Completeness: 92.11\n",
      "GCF_014596525.1 removed. Contigs: 116 N50: 106767\n",
      "GCF_014596915.1 removed. Contigs: 121 N50: 264765\n",
      "GCF_014645815.1 removed. Contigs: 129 N50: 307198\n",
      "GCF_014646055.1 removed. Contigs: 173 N50: 225409\n",
      "GCF_014646095.1 removed. Contigs: 132 N50: 247165\n",
      "GCF_014646275.1 removed. Contigs: 141 N50: 191127\n",
      "GCF_014647675.1 removed. Contigs: 108 N50: 262738\n",
      "GCF_014647695.1 removed. Contigs: 209 N50: 314084\n",
      "GCF_014647715.1 removed. Contigs: 130 N50: 73728\n",
      "GCF_014647875.1 removed. Contigs: 285 N50: 168452\n",
      "GCF_014648555.1 removed. Contigs: 125 N50: 208654\n",
      "GCF_014648615.1 removed. Contigs: 145 N50: 357856\n",
      "GCF_014648655.1 removed. Contigs: 244 N50: 195965\n",
      "GCF_014648695.1 removed. Contigs: 165 N50: 201243\n",
      "GCF_014648795.1 removed. Contigs: 130 N50: 132417\n",
      "GCF_014648915.1 removed. Contigs: 103 N50: 258264\n",
      "GCF_014649015.1 removed. Contigs: 221 N50: 388913\n",
      "GCF_014649035.1 removed. Contigs: 102 N50: 183557\n",
      "GCF_014649235.1 removed. Contigs: 134 N50: 369678\n",
      "GCF_014649475.1 removed. Contigs: 128 N50: 243992\n",
      "GCF_014649495.1 removed. Contigs: 101 N50: 364515\n",
      "GCF_014649515.1 removed. Contigs: 139 N50: 465971\n",
      "GCF_014649535.1 removed. Contigs: 114 N50: 237536\n",
      "GCF_014649595.1 removed. Contigs: 132 N50: 348767\n",
      "GCF_014649635.1 removed. Contigs: 179 N50: 253435\n",
      "GCF_014649655.1 removed. Contigs: 181 N50: 194407\n",
      "GCF_014649775.1 removed. Contigs: 167 N50: 346887\n",
      "GCF_014649795.1 removed. Contigs: 102 N50: 374786\n",
      "GCF_014649815.1 removed. Contigs: 359 N50: 306986\n",
      "GCF_014649975.1 removed. Contigs: 118 N50: 347702\n",
      "GCF_014650115.1 removed. Contigs: 135 N50: 278653\n",
      "GCF_014650135.1 removed. Contigs: 241 N50: 512933\n",
      "GCF_014650435.1 removed. Contigs: 139 N50: 452485\n",
      "GCF_014650495.1 removed. Contigs: 109 N50: 263071\n",
      "GCF_014650595.1 removed. Contigs: 179 N50: 228177\n",
      "GCF_014650615.1 removed. Contigs: 111 N50: 245055\n",
      "GCF_014650635.1 removed. Contigs: 179 N50: 221021\n",
      "GCF_014650655.1 removed. Contigs: 128 N50: 327315\n",
      "GCF_014650735.1 removed. Contigs: 272 N50: 185453\n",
      "GCF_014650835.1 removed. Contigs: 115 N50: 250279\n",
      "GCF_014650855.1 removed. Contigs: 169 N50: 164302\n",
      "GCF_014650875.1 removed. Contigs: 296 N50: 140288\n",
      "GCF_014650935.1 removed. Contigs: 236 N50: 238207\n",
      "GCF_014650955.1 removed. Contigs: 105 N50: 239402\n",
      "GCF_014650995.1 removed. Contigs: 153 N50: 263771\n",
      "GCF_014651075.1 removed. Contigs: 109 N50: 341741\n",
      "GCF_014651115.1 removed. Contigs: 108 N50: 226038\n",
      "GCF_014651135.1 removed. Contigs: 108 N50: 181340\n",
      "GCF_014651175.1 removed. Contigs: 165 N50: 524601\n",
      "GCF_014651195.1 removed. Contigs: 104 N50: 328976\n",
      "GCF_014653855.1 removed. Contigs: 163 N50: 190010\n",
      "GCF_014654675.1 removed. Contigs: 176 N50: 150732\n",
      "GCF_014654935.1 removed. Contigs: 147 N50: 257073\n",
      "GCF_014655475.1 removed. Contigs: 150 N50: 198945\n",
      "GCF_014655955.1 removed. Contigs: 198 N50: 131717\n",
      "GCF_014656075.1 removed. Contigs: 332 N50: 59557\n",
      "GCF_014656095.1 removed. Contigs: 354 N50: 45204\n",
      "GCF_014656295.1 removed. Contigs: 114 N50: 315753\n",
      "GCF_014712245.1 removed. Contigs: 265 N50: 139555\n",
      "GCF_014748275.1 removed. Contigs: 344 N50: 431047\n",
      "GCF_014852645.1 removed. Contigs: 1180 N50: 9931\n",
      "GCF_015475845.1 removed. Contigs: 119 N50: 212980\n",
      "GCF_015645455.1 removed. Contigs: 144 N50: 184031\n",
      "GCF_015689475.1 removed. Contamination: 5.26 Completeness: 100.0\n",
      "GCF_015690355.1 removed. Contamination: 6.84 Completeness: 99.47\n",
      "GCF_015912395.1 removed. Contigs: 1688 N50: 9549\n",
      "GCF_015912435.1 removed. Contigs: 1938 N50: 8002\n",
      "GCF_015912455.1 removed. Contigs: 1196 N50: 9769\n",
      "GCF_015912465.1 removed. Contigs: 1242 N50: 9241\n",
      "GCF_015912875.1 removed. Contigs: 1712 N50: 9030\n",
      "GCF_015912895.1 removed. Contigs: 1935 N50: 7660\n",
      "GCF_015912925.1 removed. Contigs: 1857 N50: 8743\n",
      "GCF_015912945.1 removed. Contigs: 1850 N50: 8414\n",
      "GCF_015912975.1 removed. Contigs: 1190 N50: 9618\n",
      "GCF_015912985.1 removed. Contigs: 1341 N50: 8658\n",
      "GCF_015912995.1 removed. Contigs: 1163 N50: 9656\n",
      "GCF_015999245.1 removed. Contigs: 111 N50: 995929\n",
      "GCF_016017415.1 removed. Contigs: 331 N50: 41623\n",
      "GCF_016103465.1 removed. Contigs: 1327 N50: 15605\n",
      "GCF_016103505.1 removed. Contamination: 12.654135338345869 Completeness: 94.26315789473684\n",
      "GCF_016107395.1 removed. Contigs: 475 N50: 152116\n",
      "GCF_016464295.1 removed. Contigs: 240 N50: 82658\n",
      "GCF_016491645.1 removed. Contigs: 1164 N50: 17257\n",
      "GCF_016598475.1 removed. Contigs: 352 N50: 83341\n",
      "GCF_016612545.1 removed. Contigs: 833 N50: 30177\n",
      "GCF_016612575.1 removed. Contigs: 602 N50: 24967\n",
      "GCF_016612605.1 removed. Contigs: 497 N50: 30897\n",
      "GCF_016612625.1 removed. Contigs: 757 N50: 31686\n",
      "GCF_016612645.1 removed. Contigs: 598 N50: 25388\n",
      "GCF_016612655.1 removed. Contigs: 312 N50: 69644\n",
      "GCF_016612665.1 removed. Contigs: 552 N50: 26977\n",
      "GCF_016612705.1 removed. Contigs: 589 N50: 27955\n",
      "GCF_016612725.1 removed. Contigs: 498 N50: 30984\n",
      "GCF_016612755.1 removed. Contigs: 357 N50: 41187\n",
      "GCF_016612775.1 removed. Contigs: 635 N50: 24564\n",
      "GCF_016612805.1 removed. Contigs: 1146 N50: 14743\n",
      "GCF_016612825.1 removed. Contigs: 762 N50: 24257\n",
      "GCF_016612855.1 removed. Contigs: 537 N50: 41669\n",
      "GCF_016613285.1 removed. Contigs: 134 N50: 105553\n",
      "GCF_016613335.1 removed. Contigs: 126 N50: 119828\n",
      "GCF_016653995.1 removed. Contigs: 622 N50: 27814\n",
      "GCF_016654055.1 removed. Contigs: 591 N50: 26144\n",
      "GCF_016654065.1 removed. Contigs: 597 N50: 25060\n",
      "GCF_016654115.1 removed. Contigs: 543 N50: 43101\n",
      "GCF_016741775.1 removed. Contigs: 106 N50: 270358\n",
      "GCF_016741855.1 removed. Contigs: 148 N50: 398829\n",
      "GCF_016741875.1 removed. Contigs: 253 N50: 229793\n",
      "GCF_016741935.1 removed. Contigs: 269 N50: 328265\n",
      "GCF_016745505.1 removed. Contigs: 207 N50: 64915\n",
      "GCF_016757295.1 removed. Contigs: 369 N50: 99471\n",
      "GCF_016834575.1 removed. Contigs: 214 N50: 65020\n",
      "GCF_016901035.1 removed. Contigs: 616 N50: 26787\n",
      "GCF_016905445.1 removed. Contigs: 583 N50: 33264\n",
      "GCF_016921115.1 removed. Contigs: 306 N50: 190774\n",
      "GCF_017114865.1 removed. Contigs: 105 N50: 171252\n",
      "GCF_017164735.1 removed. Contigs: 715 N50: 16285\n",
      "GCF_017315755.1 removed. Contigs: 1769 N50: 21718\n",
      "GCF_017353455.1 removed. Contamination: 0.79 Completeness: 85.61\n",
      "GCF_017427025.1 removed. Contigs: 1261 N50: 10071\n",
      "GCF_017592595.1 removed. Contigs: 135 N50: 272704\n",
      "GCF_017614965.1 removed. Contigs: 773 N50: 25652\n",
      "GCF_017676365.1 removed. Contigs: 405 N50: 31324\n",
      "GCF_017813245.1 removed. Contigs: 402 N50: 35783\n",
      "GCF_017948485.1 removed. Contigs: 117 N50: 299478\n",
      "GCF_018035285.1 removed. Contigs: 893 N50: 18372\n",
      "GCF_018070205.1 removed. Contigs: 143 N50: 171449\n",
      "GCF_018070535.1 removed. Contigs: 126 N50: 202649\n",
      "GCF_018070565.1 removed. Contigs: 134 N50: 164477\n",
      "GCF_018070605.1 removed. Contigs: 245 N50: 66869\n",
      "GCF_018070745.1 removed. Contigs: 171 N50: 134500\n",
      "GCF_018555775.1 removed. Contigs: 731 N50: 22476\n",
      "GCF_018614005.1 removed. Contigs: 102 N50: 149180\n",
      "GCF_018614035.1 removed. Contigs: 101 N50: 139003\n",
      "GCF_018614055.1 removed. Contigs: 122 N50: 165028\n",
      "GCF_018614075.1 removed. Contigs: 149 N50: 74083\n",
      "GCF_018614095.1 removed. Contigs: 155 N50: 81888\n",
      "GCF_018614105.1 removed. Contigs: 197 N50: 57180\n",
      "GCF_018614135.1 removed. Contigs: 250 N50: 42985\n",
      "GCF_018614155.1 removed. Contigs: 196 N50: 59624\n",
      "GCF_018614185.1 removed. Contigs: 104 N50: 130389\n",
      "GCF_018614195.1 removed. Contigs: 127 N50: 142553\n",
      "GCF_018614205.1 removed. Contigs: 216 N50: 55601\n",
      "GCF_018614355.1 removed. Contigs: 133 N50: 107847\n",
      "GCF_018614415.1 removed. Contigs: 217 N50: 60696\n",
      "GCF_018614445.1 removed. Contigs: 117 N50: 194125\n",
      "GCF_018614475.1 removed. Contigs: 152 N50: 88780\n",
      "GCF_018614515.1 removed. Contigs: 102 N50: 202061\n",
      "GCF_018614525.1 removed. Contigs: 219 N50: 43578\n",
      "GCF_018614585.1 removed. Contamination: 0.0 Completeness: 81.58\n",
      "GCF_018614615.1 removed. Contigs: 154 N50: 111385\n",
      "GCF_018619165.1 removed. Contigs: 1524 N50: 7331\n",
      "GCF_018619185.1 removed. Contigs: 586 N50: 23710\n",
      "GCF_018619195.1 removed. Contigs: 1079 N50: 11823\n",
      "GCF_018619235.1 removed. Contigs: 1189 N50: 8953\n",
      "GCF_018619575.1 removed. Contigs: 101 N50: 143393\n",
      "GCF_018638295.1 removed. Contigs: 206 N50: 204147\n",
      "GCF_018638305.1 removed. Contigs: 210 N50: 76195\n",
      "GCF_018638335.1 removed. Contigs: 179 N50: 98291\n",
      "GCF_018883605.1 removed. Contigs: 1379 N50: 17460\n",
      "GCF_018927715.1 removed. Contigs: 898 N50: 43214\n",
      "GCF_018982965.1 removed. Contigs: 286 N50: 57920\n",
      "GCF_019049085.1 removed. Contigs: 256 N50: 109083\n",
      "GCF_019049125.1 removed. Contigs: 139 N50: 203454\n",
      "GCF_019049165.1 removed. Contamination: 16.849529780564264 Completeness: 100.0\n",
      "GCF_019049185.1 removed. Contigs: 175 N50: 110869\n",
      "GCF_019049205.1 removed. Contigs: 204 N50: 137950\n",
      "GCF_019049225.1 removed. Contigs: 157 N50: 167824\n",
      "GCF_019049245.1 removed. Contigs: 243 N50: 201133\n",
      "GCF_019049255.1 removed. Contigs: 232 N50: 216656\n",
      "GCF_019049285.1 removed. Contigs: 219 N50: 151084\n",
      "GCF_019049305.1 removed. Contigs: 259 N50: 93530\n",
      "GCF_019049325.1 removed. Contigs: 139 N50: 217020\n",
      "GCF_019049345.1 removed. Contamination: 104.3103448275862 Completeness: 100.0\n",
      "GCF_019090125.1 removed. Contigs: 699 N50: 28611\n",
      "GCF_019090165.1 removed. Contigs: 212 N50: 114510\n",
      "GCF_019091445.1 removed. Contigs: 840 N50: 229965\n",
      "GCF_019091475.1 removed. Contigs: 568 N50: 212700\n",
      "GCF_019104725.1 removed. Contigs: 111 N50: 109407\n",
      "GCF_019192735.1 removed. Contigs: 209 N50: 131431\n",
      "GCF_019219825.1 removed. Contigs: 153 N50: 222188\n",
      "GCF_019303475.1 removed. Contigs: 118 N50: 98878\n",
      "GCF_019350955.1 removed. Contigs: 1759 N50: 7641\n",
      "GCF_019399205.1 removed. Contigs: 653 N50: 29528\n",
      "GCF_019399235.1 removed. Contigs: 695 N50: 26939\n",
      "GCF_019400875.1 removed. Contigs: 881 N50: 13615\n",
      "GCF_019400985.1 removed. Contigs: 763 N50: 20623\n",
      "GCF_019456635.1 removed. Contigs: 373 N50: 100604\n",
      "GCF_019641555.1 removed. Contigs: 1020 N50: 17661\n",
      "GCF_019710745.1 removed. Contigs: 212 N50: 148377\n",
      "GCF_019733235.1 removed. Contigs: 106 N50: 281159\n",
      "GCF_019733245.1 removed. Contigs: 441 N50: 35025\n",
      "GCF_019857225.1 removed. Contigs: 190 N50: 119887\n",
      "GCF_019890635.1 removed. Contigs: 150 N50: 95602\n",
      "GCF_019890645.1 removed. Contigs: 174 N50: 157308\n",
      "GCF_019890725.1 removed. Contigs: 171 N50: 116494\n",
      "GCF_020024005.1 removed. Contigs: 494 N50: 23973\n",
      "GCF_020037025.1 removed. Contigs: 587 N50: 41451\n",
      "GCF_020103755.1 removed. Contigs: 1295 N50: 13306\n",
      "GCF_020166595.1 removed. Contigs: 1079 N50: 14701\n",
      "GCF_020176795.1 removed. Contigs: 991 N50: 15698\n",
      "GCF_020177475.1 removed. Contigs: 1018 N50: 12009\n",
      "GCF_020181035.1 removed. Contigs: 104 N50: 352661\n",
      "GCF_020312235.1 removed. Contigs: 120 N50: 129602\n",
      "GCF_020312255.1 removed. Contigs: 129 N50: 97815\n",
      "GCF_020312265.1 removed. Contigs: 173 N50: 90653\n",
      "GCF_020312315.1 removed. Contigs: 112 N50: 113910\n",
      "GCF_020328095.1 removed. Contigs: 297 N50: 62247\n",
      "GCF_020400585.1 removed. Contigs: 112 N50: 269769\n",
      "GCF_020400595.1 removed. Contigs: 216 N50: 193034\n",
      "GCF_020400605.1 removed. Contigs: 137 N50: 230881\n",
      "GCF_020400685.1 removed. Contigs: 113 N50: 400654\n",
      "GCF_020400725.1 removed. Contigs: 229 N50: 78472\n",
      "GCF_020400825.1 removed. Contigs: 188 N50: 203412\n",
      "GCF_020404845.1 removed. Contigs: 169 N50: 130065\n",
      "GCF_020521255.1 removed. Contamination: 5.24 Completeness: 99.58\n",
      "GCF_020521295.1 removed. Contamination: 6.4 Completeness: 98.87\n",
      "GCF_020532625.1 removed. Contigs: 284 N50: 38913\n",
      "GCF_020532645.1 removed. Contigs: 169 N50: 106707\n",
      "GCF_020564935.1 removed. Contigs: 133 N50: 117612\n",
      "GCF_020685565.1 removed. Contigs: 106 N50: 362883\n",
      "GCF_020740835.1 removed. Contigs: 1076 N50: 12946\n",
      "GCF_020740895.1 removed. Contigs: 792 N50: 17091\n",
      "GCF_020741675.1 removed. Contigs: 148 N50: 80118\n",
      "GCF_020819595.1 removed. Contigs: 141 N50: 364153\n",
      "GCF_020861705.1 removed. Contigs: 113 N50: 237932\n",
      "GCF_020873855.1 removed. Contigs: 159 N50: 490332\n",
      "GCF_020873925.1 removed. Contigs: 102 N50: 255674\n",
      "GCF_020881015.1 removed. Contigs: 440 N50: 214941\n",
      "GCF_020907985.1 removed. Contigs: 150 N50: 134447\n",
      "GCF_020991365.1 removed. Contigs: 200 N50: 217664\n",
      "GCF_021083325.1 removed. Contigs: 182 N50: 69524\n",
      "GCF_021228125.1 removed. Contigs: 108 N50: 326773\n",
      "GCF_021228305.1 removed. Contigs: 115 N50: 523938\n",
      "GCF_021233395.1 removed. Contamination: 5.26 Completeness: 100.0\n",
      "GCF_021246025.1 removed. Contamination: 5.79 Completeness: 100.0\n",
      "GCF_021300715.1 removed. Contigs: 289 N50: 55964\n",
      "GCF_021366495.1 removed. Contigs: 142 N50: 183934\n",
      "GCF_021432205.1 removed. Contigs: 1133 N50: 15125\n",
      "GCF_021432215.1 removed. Contigs: 1377 N50: 10652\n",
      "GCF_021432225.1 removed. Contigs: 1214 N50: 12424\n",
      "GCF_021432285.1 removed. Contigs: 1245 N50: 12220\n",
      "GCF_021442405.1 removed. Contigs: 254 N50: 47382\n",
      "GCF_021442415.1 removed. Contigs: 296 N50: 42201\n",
      "GCF_021462265.1 removed. Contigs: 253 N50: 185809\n",
      "GCF_021462825.1 removed. Contigs: 125 N50: 202410\n",
      "GCF_021474405.1 removed. Contigs: 309 N50: 50087\n",
      "GCF_021474425.1 removed. Contigs: 670 N50: 31546\n",
      "GCF_021538775.1 removed. Contigs: 116 N50: 289470\n",
      "GCF_021538815.1 removed. Contigs: 291 N50: 49328\n",
      "GCF_021538835.1 removed. Contigs: 401 N50: 37309\n",
      "GCF_021538875.1 removed. Contigs: 391 N50: 40525\n",
      "GCF_021538895.1 removed. Contigs: 281 N50: 70511\n",
      "GCF_021545335.1 removed. Contamination: 1.07 Completeness: 75.86\n",
      "GCF_021556455.1 removed. Contigs: 811 N50: 12419\n",
      "GCF_021696945.1 removed. Contigs: 219 N50: 63493\n",
      "GCF_021890875.1 removed. Contigs: 271 N50: 86830\n",
      "GCF_021892415.1 removed. Contigs: 586 N50: 33554\n",
      "GCF_022015835.1 removed. Contigs: 596 N50: 16807\n",
      "GCF_022213265.1 removed. Contigs: 118 N50: 151133\n",
      "GCF_022288725.1 removed. Contamination: 9.06 Completeness: 98.95\n",
      "GCF_022347745.1 removed. Contigs: 638 N50: 20066\n",
      "GCF_022392825.1 removed. Contamination: 6.01 Completeness: 99.93\n",
      "GCF_022414535.1 removed. Contigs: 134 N50: 127370\n",
      "GCF_022414575.1 removed. Contigs: 128 N50: 107719\n",
      "GCF_022414595.1 removed. Contigs: 147 N50: 84552\n",
      "GCF_022414615.1 removed. Contigs: 164 N50: 164408\n",
      "GCF_022414635.1 removed. Contigs: 169 N50: 100013\n",
      "GCF_022614595.1 removed. Contigs: 225 N50: 27697\n",
      "GCF_022695985.1 removed. Contigs: 461 N50: 29485\n",
      "GCF_023072675.1 removed. Contigs: 720 N50: 18780\n",
      "GCF_023072755.1 removed. Contigs: 422 N50: 26659\n",
      "GCF_023155275.1 removed. Contigs: 117 N50: 120589\n",
      "GCF_023370675.1 removed. Contigs: 257 N50: 227024\n",
      "GCF_023502025.1 removed. Contigs: 107 N50: 189032\n",
      "GCF_023502165.1 removed. Contigs: 114 N50: 126951\n",
      "GCF_023516595.1 removed. Contigs: 271 N50: 152317\n",
      "GCF_023516615.1 removed. Contigs: 363 N50: 71020\n",
      "GCF_023534765.1 removed. Contigs: 102 N50: 194184\n",
      "GCF_023544705.1 removed. Contigs: 138 N50: 131274\n",
      "GCF_023554615.1 removed. Contigs: 484 N50: 228026\n",
      "GCF_023630175.1 removed. Contigs: 142 N50: 122618\n",
      "GCF_023630275.1 removed. Contigs: 278 N50: 206944\n",
      "GCF_023667545.1 removed. Contigs: 493 N50: 26602\n",
      "GCF_023669525.1 removed. Contamination: 60.736677115987455 Completeness: 82.44514106583073\n",
      "GCF_023701525.1 removed. Contigs: 138 N50: 266232\n",
      "GCF_023713605.1 removed. Contigs: 176 N50: 243441\n",
      "GCF_023717205.1 removed. Contigs: 105 N50: 145714\n",
      "GCF_023717255.1 removed. Contamination: 5.79 Completeness: 99.47\n",
      "GCF_023805375.1 removed. Contigs: 177 N50: 116947\n",
      "GCF_023887685.1 removed. Contigs: 137 N50: 163816\n",
      "GCF_023930305.1 removed. Contigs: 129 N50: 117347\n",
      "GCF_024105145.1 removed. Contigs: 341 N50: 97324\n",
      "GCF_024105195.1 removed. Contigs: 575 N50: 131685\n",
      "GCF_024105205.1 removed. Contigs: 323 N50: 108708\n",
      "GCF_024105315.1 removed. Contigs: 169 N50: 146186\n",
      "GCF_024105335.1 removed. Contigs: 467 N50: 152463\n",
      "GCF_024107415.1 removed. Contigs: 421 N50: 148232\n",
      "GCF_024107435.1 removed. Contigs: 322 N50: 156924\n",
      "GCF_024107455.1 removed. Contigs: 198 N50: 170996\n",
      "GCF_024107465.1 removed. Contigs: 297 N50: 177678\n",
      "GCF_024107495.1 removed. Contigs: 503 N50: 204393\n",
      "GCF_024107515.1 removed. Contigs: 128 N50: 251296\n",
      "GCF_024138375.1 removed. Contigs: 392 N50: 40528\n",
      "GCF_024223055.1 removed. Contamination: 7.01 Completeness: 98.93\n",
      "GCF_024380135.1 removed. Contigs: 148 N50: 117468\n",
      "GCF_024435335.1 removed. Contigs: 181 N50: 91403\n",
      "GCF_024435355.1 removed. Contigs: 119 N50: 323159\n",
      "GCF_024436055.1 removed. Contamination: 6.84 Completeness: 100.0\n",
      "GCF_024448155.1 removed. Contigs: 378 N50: 39005\n",
      "GCF_024448165.1 removed. Contigs: 443 N50: 38974\n",
      "GCF_024448175.1 removed. Contigs: 354 N50: 34753\n",
      "GCF_024505145.1 removed. Contigs: 298 N50: 183891\n",
      "GCF_024505305.1 removed. Contigs: 107 N50: 305657\n",
      "GCF_024519315.1 removed. Contigs: 306 N50: 119153\n",
      "GCF_024519335.1 removed. Contigs: 110 N50: 173799\n",
      "GCF_024519455.1 removed. Contigs: 130 N50: 295667\n",
      "GCF_024539675.1 removed. Contigs: 949 N50: 44277\n",
      "GCF_024586905.1 removed. Contigs: 126 N50: 137629\n",
      "GCF_024721315.2 removed. Contigs: 408 N50: 45434\n",
      "GCF_024721345.1 removed. Contigs: 308 N50: 57957\n",
      "GCF_025153975.1 removed. Contigs: 113 N50: 158519\n",
      "GCF_025290915.1 removed. Contigs: 560 N50: 21691\n",
      "GCF_025399755.1 removed. Contigs: 585 N50: 27723\n",
      "GCF_025399765.1 removed. Contigs: 321 N50: 215380\n",
      "GCF_025399775.1 removed. Contigs: 320 N50: 69118\n",
      "GCF_025399795.1 removed. Contigs: 244 N50: 147426\n",
      "GCF_025568765.1 removed. Contigs: 156 N50: 214672\n",
      "GCF_025656915.1 removed. Contigs: 420 N50: 31411\n",
      "GCF_025716995.1 removed. Contigs: 619 N50: 22387\n",
      "GCF_025964845.1 removed. Contigs: 655 N50: 17082\n",
      "GCF_026008495.1 removed. Contigs: 228 N50: 163876\n",
      "GCF_026126445.1 removed. Contigs: 103 N50: 229202\n",
      "GCF_026168105.1 removed. Contigs: 159 N50: 151047\n",
      "GCF_026183975.1 removed. Contigs: 122 N50: 273762\n",
      "GCF_026184015.1 removed. Contigs: 109 N50: 165478\n",
      "GCF_026191925.1 removed. Contigs: 910 N50: 15672\n",
      "GCF_026242095.1 removed. Contigs: 107 N50: 260136\n",
      "GCF_026262575.1 removed. Contigs: 141 N50: 170825\n",
      "GCF_026274175.1 removed. Contigs: 342 N50: 37912\n",
      "GCF_026341425.1 removed. Contigs: 102 N50: 163682\n",
      "GCF_026420845.1 removed. Contigs: 105 N50: 186895\n",
      "GCF_026627185.1 removed. Contigs: 160 N50: 128302\n",
      "GCF_026627225.1 removed. Contigs: 274 N50: 49007\n",
      "GCF_026627245.1 removed. Contigs: 186 N50: 79368\n",
      "GCF_026627445.1 removed. Contigs: 176 N50: 150361\n",
      "GCF_026627485.1 removed. Contigs: 448 N50: 61903\n",
      "GCF_026627505.1 removed. Contigs: 380 N50: 79555\n",
      "GCF_026800165.1 removed. Contigs: 1345 N50: 55179\n",
      "GCF_026965885.1 removed. Contigs: 151 N50: 88305\n",
      "GCF_027157945.1 removed. Contigs: 303 N50: 64321\n",
      "GCF_027158125.1 removed. Contigs: 306 N50: 89003\n",
      "GCF_027158205.1 removed. Contigs: 146 N50: 156184\n",
      "GCF_027270795.1 removed. Contigs: 1048 N50: 19949\n",
      "GCF_027270915.1 removed. Contigs: 684 N50: 30252\n",
      "GCF_027270975.1 removed. Contigs: 964 N50: 14104\n",
      "GCF_027533845.1 removed. Contigs: 139 N50: 304118\n",
      "GCF_027570385.1 removed. Contigs: 241 N50: 109216\n",
      "GCF_027920075.1 removed. Contigs: 114 N50: 288023\n",
      "GCF_027947535.2 removed. Contamination: 5.307017543859649 Completeness: 99.47368421052632\n",
      "GCF_027947545.2 removed. Contigs: 112 N50: 180908\n",
      "GCF_028369295.1 removed. Contigs: 200 N50: 146433\n",
      "GCF_028401405.1 removed. Contigs: 256 N50: 148220\n",
      "GCF_028657195.3 removed. Contigs: 125 N50: 158355\n",
      "GCF_028863365.1 removed. Contigs: 137 N50: 339391\n",
      "GCF_029223525.1 removed. Contamination: 7.280701754385965 Completeness: 100.0\n",
      "GCF_029269095.1 removed. Contigs: 1671 N50: 12630\n",
      "GCF_029269835.1 removed. Contigs: 111 N50: 345819\n",
      "GCF_029813775.1 removed. Contigs: 164 N50: 147138\n",
      "GCF_029813835.1 removed. Contigs: 283 N50: 73719\n",
      "GCF_029846675.1 removed. Contigs: 116 N50: 146088\n",
      "GCF_029846775.1 removed. Contigs: 146 N50: 127268\n",
      "GCF_029892105.1 removed. Contigs: 138 N50: 200501\n",
      "GCF_029892145.1 removed. Contigs: 110 N50: 169698\n",
      "GCF_029894335.1 removed. Contigs: 118 N50: 135651\n",
      "GCF_029910965.1 removed. Contigs: 354 N50: 56750\n",
      "GCF_029946625.1 removed. Contigs: 586 N50: 25306\n",
      "GCF_029953285.1 removed. Contigs: 114 N50: 152462\n",
      "GCF_029967595.1 removed. Contigs: 264 N50: 58042\n",
      "GCF_029967605.1 removed. Contigs: 384 N50: 36481\n",
      "GCF_029968015.1 removed. Contigs: 113 N50: 164732\n",
      "GCF_030010495.1 removed. Contigs: 605 N50: 30402\n",
      "GCF_030010595.1 removed. Contigs: 276 N50: 67870\n",
      "GCF_030010605.1 removed. Contigs: 135 N50: 246320\n",
      "GCF_030010625.1 removed. Contigs: 129 N50: 149073\n",
      "GCF_030010655.1 removed. Contigs: 212 N50: 99634\n",
      "GCF_030010675.1 removed. Contigs: 791 N50: 19547\n",
      "GCF_030056615.1 removed. Contigs: 135 N50: 116958\n",
      "GCF_030063135.1 removed. Contigs: 140 N50: 129914\n",
      "GCF_030063325.1 removed. Contigs: 192 N50: 117472\n",
      "GCF_030063355.1 removed. Contigs: 237 N50: 81534\n",
      "GCF_030063415.1 removed. Contigs: 128 N50: 165176\n",
      "GCF_030122705.1 removed. Contamination: 5.2631578947368425 Completeness: 100.0\n",
      "GCF_030159695.1 removed. Contigs: 114 N50: 584103\n",
      "GCF_030238025.1 removed. Contigs: 166 N50: 108182\n",
      "GCF_030262575.1 removed. Contigs: 218 N50: 165811\n",
      "GCF_030268605.1 removed. Contigs: 197 N50: 244339\n",
      "GCF_030268625.1 removed. Contigs: 179 N50: 321264\n",
      "GCF_030268665.1 removed. Contigs: 340 N50: 165723\n",
      "GCF_030268805.1 removed. Contigs: 102 N50: 321888\n",
      "GCF_030268825.1 removed. Contigs: 221 N50: 138939\n",
      "GCF_030268885.1 removed. Contigs: 169 N50: 221631\n",
      "GCF_030270785.1 removed. Contigs: 187 N50: 189630\n",
      "GCF_900090095.1 removed. Contigs: 106 N50: 261762\n",
      "GCF_900091315.1 removed. Contigs: 243 N50: 70678\n",
      "GCF_900091485.1 removed. Contigs: 193 N50: 55778\n",
      "GCF_900091715.1 removed. Contigs: 239 N50: 56194\n",
      "GCF_900091725.1 removed. Contigs: 102 N50: 266092\n",
      "GCF_900091745.1 removed. Contigs: 118 N50: 133137\n",
      "GCF_900091755.1 removed. Contigs: 377 N50: 32343\n",
      "GCF_900091765.1 removed. Contigs: 301 N50: 38665\n",
      "GCF_900091785.1 removed. Contigs: 352 N50: 35237\n",
      "GCF_900091815.1 removed. Contigs: 652 N50: 18519\n",
      "GCF_900091825.1 removed. Contigs: 1185 N50: 7383\n",
      "GCF_900091835.1 removed. Contigs: 190 N50: 80767\n",
      "GCF_900091845.1 removed. Contigs: 400 N50: 38736\n",
      "GCF_900091855.1 removed. Contigs: 670 N50: 21543\n",
      "GCF_900091865.1 removed. Contigs: 253 N50: 52422\n",
      "GCF_900091875.1 removed. Contigs: 336 N50: 35127\n",
      "GCF_900091885.1 removed. Contigs: 157 N50: 88536\n",
      "GCF_900091905.1 removed. Contigs: 1456 N50: 8512\n",
      "GCF_900091955.1 removed. Contigs: 244 N50: 84953\n",
      "GCF_900091975.1 removed. Contigs: 242 N50: 72318\n",
      "GCF_900091985.1 removed. Contigs: 420 N50: 35698\n",
      "GCF_900091995.1 removed. Contigs: 1302 N50: 9389\n",
      "GCF_900092005.1 removed. Contigs: 176 N50: 91291\n",
      "GCF_900092015.1 removed. Contamination: 5.79 Completeness: 98.82\n",
      "GCF_900095575.1 removed. Contigs: 111 N50: 129868\n",
      "GCF_900110255.1 removed. Contigs: 112 N50: 183757\n",
      "GCF_900110755.1 removed. Contigs: 122 N50: 143556\n",
      "GCF_900114955.1 removed. Contigs: 106 N50: 302548\n",
      "GCF_900119365.1 removed. Contigs: 103 N50: 269459\n",
      "GCF_900171555.1 removed. Contigs: 334 N50: 48334\n",
      "GCF_900290235.1 removed. Contigs: 192 N50: 142982\n",
      "GCF_902506575.1 removed. Contigs: 159 N50: 7159421\n",
      "GCF_907177275.1 removed. Contigs: 135 N50: 178610\n",
      "GCF_913919575.1 removed. Contigs: 211 N50: 78904\n",
      "GCF_940670765.1 removed. Contigs: 144 N50: 235910\n",
      "GCF_945929885.1 removed. Contigs: 126 N50: 100724\n",
      "NBC_00217 removed. Contamination: 15.879828326180258 Completeness: 100.0\n",
      "NBC_00359 removed. Contamination: 5.181347150259067 Completeness: 100.0\n",
      "NBC_00845 removed. Contamination: 5.175438596491229 Completeness: 100.0\n",
      "NBC_00887 removed. Contamination: 23.6061684460261 Completeness: 100.0\n",
      "NBC_00899 removed. Contamination: 28.157894736842103 Completeness: 99.47368421052632\n",
      "NBC_01320 removed. Contamination: 5.160142348754448 Completeness: 100.0\n",
      "NBC_01484 removed. Contamination: 6.476683937823834 Completeness: 100.0\n",
      "NBC_01499 removed. Contamination: 5.027978496023608 Completeness: 99.74937343358395\n",
      "NBC_01510 removed. Contamination: 5.872193436960276 Completeness: 100.0\n",
      "NBC_01555 removed. Contamination: 5.175438596491229 Completeness: 100.0\n"
     ]
    }
   ],
   "source": [
    "df = pd.DataFrame(index=df_gtdb_meta.index, columns=[\"genome_id\", \"genus\", \"source\", \"species\", \"quality\", \"completeness\", \"contamination\", \"N50\", \"contigs\", \"genome_len\"])\n",
    "df[\"genome_id\"] = df.index\n",
    "\n",
    "\n",
    "for genome_id in df.index:\n",
    "    # Define genus\n",
    "    if genome_id in df_gtdb_meta.index:\n",
    "        genus = df_gtdb_meta.loc[genome_id, \"Genus\"].split(\"__\")[1]\n",
    "        species = df_gtdb_meta.loc[genome_id, \"Species\"]\n",
    "        \n",
    "        df.loc[genome_id, \"species\"] = species\n",
    "        df.loc[genome_id, \"genus\"] = genus\n",
    "    else:\n",
    "        print(genome_id, \"not found in df_gtdb_meta!\")\n",
    "        \n",
    "    # Define quality levels\n",
    "    if genome_id in df_ncbi_meta_1.index:\n",
    "        df.loc[genome_id, \"source\"] = \"NCBI\"\n",
    "        assembly_level = df_ncbi_meta_1.loc[genome_id, \"assembly_level\"]\n",
    "    else:\n",
    "        df.loc[genome_id, \"source\"] = \"NBC\"\n",
    "        if genome_id in df_nbc_assembly_level.index:\n",
    "            assembly_level = df_nbc_assembly_level.loc[genome_id, \"assembly_level\"]\n",
    "        else:\n",
    "            assembly_level = \"Complete Genome\"\n",
    "\n",
    "            \n",
    "    # Define checkM metrics of completeness and contamination\n",
    "    if genome_id in df_checkm_meta_2.index:\n",
    "        completeness = df_checkm_meta_2.loc[genome_id, \"Completeness\"]\n",
    "        contamination = df_checkm_meta_2.loc[genome_id, \"Contamination\"]\n",
    "    else:\n",
    "        completeness = df_gtdb_meta.loc[genome_id, \"checkm_completeness\"]\n",
    "        contamination = df_gtdb_meta.loc[genome_id, \"checkm_contamination\"]\n",
    "\n",
    "    df.loc[genome_id, 'completeness'] = completeness\n",
    "    df.loc[genome_id, 'contamination'] = contamination\n",
    "\n",
    "    # Define seqfu metrics of N50 and Contigs\n",
    "    df.loc[genome_id, 'N50'] = df_seqfu_meta.loc[genome_id, \"N50\"]\n",
    "    df.loc[genome_id, 'contigs'] = df_seqfu_meta.loc[genome_id, \"Count\"]\n",
    "    df.loc[genome_id, 'genome_len'] = df_seqfu_meta.loc[genome_id, \"Total\"]\n",
    "    df.loc[genome_id, 'gc'] = df_seqfu_meta.loc[genome_id, \"gc\"]\n",
    "\n",
    "    # Define the quality of genomes\n",
    "    if contamination > 5 or completeness < 90:\n",
    "        df.loc[genome_id, \"quality\"] = \"LQ\"\n",
    "        print(genome_id, \"removed.\", \"Contamination:\", contamination, \"Completeness:\",completeness)\n",
    "    else:\n",
    "        if assembly_level in [\"Complete Genome\", \"Chromosome\"]:\n",
    "            df.loc[genome_id, \"quality\"] = \"HQ\"\n",
    "        else:\n",
    "            contigs = df_seqfu_meta.loc[genome_id, \"Count\"]\n",
    "            N50 = df_seqfu_meta.loc[genome_id, \"N50\"]\n",
    "            if contigs <= 100 and  N50 >= 100000:\n",
    "                df.loc[genome_id, \"quality\"] = \"MQ\"\n",
    "            else:\n",
    "                print(genome_id, \"removed.\", \"Contigs:\", contigs, \"N50:\", N50)\n",
    "                df.loc[genome_id, \"quality\"] = \"LQ\"\n",
    "\n",
    "df_stats = df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "02cc349d-6d0e-44e2-ae09-8884839bcf81",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "quality\n",
       "HQ    1347\n",
       "LQ    1264\n",
       "MQ    1229\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_stats.quality.value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "26de514c-f23a-464b-a741-76dd4e29b888",
   "metadata": {},
   "source": [
    "# 4. Streptomyces genus medium quality project\n",
    "\n",
    "By following the above assessement, different quality filters were defined to select medium to high quality genomes for the next BGCFlow project called `mq_strepto`\n",
    "\n",
    "Low quality\n",
    "1. CheckM contamination > 5 %\n",
    "2. CheckM completeness < 90 %\n",
    "3. Number of contigs > 100\n",
    "4. N50 < 100 kb\n",
    "\n",
    "Medium quality\n",
    "1. NCBI assembly level = Cotig, Scaffold\n",
    "2. CheckM contamination <= 5 %\n",
    "3. CheckM completeness <= 90 %\n",
    "4. Number of contigs <= 100\n",
    "5. N50 >= 100 kb\n",
    "\n",
    "High quality\n",
    "1. NCBI assembly level = Complete genome, Chromosome\n",
    "2. CheckM contamination <= 5 %\n",
    "3. CheckM completeness <= 90 %\n",
    "4. Number of contigs <= 100\n",
    "5. N50 >= 100 kb\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "24d962dc-1d78-4ee1-ac04-b30d813f6ce1",
   "metadata": {},
   "source": [
    "### Setup the config project using BGCFlow wrapper\n",
    "\n",
    "Follow the guide to set up a project called \"mq_strepto\" with samples tables and project config with the following rules\n",
    "\n",
    "```\n",
    "cd bgcflow_dir\n",
    "bgcflow init --project mq_strepto\n",
    "cp .examples/_genome_project_example/prokka-db.csv config/mq_strepto/\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7fb1479c-b0dd-498f-a962-752f23851b04",
   "metadata": {},
   "source": [
    "## Update project_config\n",
    "\n",
    "Update the file `config/qc_strepto_ncbi/project_config.yaml` to allow seqfu rule on.\n",
    "\n",
    "Note that the latest release 214 of GTDB was used here with version 2.3.0 updated in the `workflow/envs/gtdbtk.yaml` file.\n",
    "\n",
    "```\n",
    "name: mq_strepto\n",
    "pep_version: 2.1.0\n",
    "description: 'Project to of 2371 genomes of medium to high quality genomes of Streptomyces genus.'\n",
    "sample_table: samples.csv\n",
    "prokka-db: 'prokka-db.csv'\n",
    "rules:\n",
    "  seqfu: true\n",
    "  mash: true\n",
    "  prokka: true\n",
    "\n",
    "#### CUSTOM RULE CONFIGURATION ####\n",
    "rule_p  install_gtdbtk:\n",
    "    release: \"214\"\n",
    "    release_version: \"214\"\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "fe63b48a-e49f-4dfe-b76a-eb36266a8145",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Extract Streptomyces genus and MQ, HQ genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "cc5603cd-1b37-45e8-98fa-db90e7f5ff2e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "quality\n",
       "HQ    1196\n",
       "MQ    1175\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_streptomyces = df_stats[df_stats.genus == \"Streptomyces\"]\n",
    "df_strepto_mq = df_streptomyces[df_streptomyces.quality.isin([\"HQ\", \"MQ\"])]\n",
    "df_strepto_mq.quality.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "9a5e7891-6cca-4415-9d4b-983e38d4c88e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source</th>\n",
       "      <th>organism</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "      <th>strain</th>\n",
       "      <th>closest_placement_reference</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genome_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GCF_000009765.2</th>\n",
       "      <td>ncbi</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000010605.1</th>\n",
       "      <td>ncbi</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000091305.1</th>\n",
       "      <td>ncbi</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000092385.1</th>\n",
       "      <td>ncbi</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000147815.2</th>\n",
       "      <td>ncbi</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NBC_01798</th>\n",
       "      <td>custom</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NBC_01798</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NBC_01799</th>\n",
       "      <td>custom</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NBC_01799</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NBC_01800</th>\n",
       "      <td>custom</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NBC_01800</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NBC_01803</th>\n",
       "      <td>custom</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NBC_01803</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NBC_01808</th>\n",
       "      <td>custom</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NBC_01808</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2371 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 source organism genus species     strain  \\\n",
       "genome_id                                                   \n",
       "GCF_000009765.2    ncbi      NaN   NaN     NaN        NaN   \n",
       "GCF_000010605.1    ncbi      NaN   NaN     NaN        NaN   \n",
       "GCF_000091305.1    ncbi      NaN   NaN     NaN        NaN   \n",
       "GCF_000092385.1    ncbi      NaN   NaN     NaN        NaN   \n",
       "GCF_000147815.2    ncbi      NaN   NaN     NaN        NaN   \n",
       "...                 ...      ...   ...     ...        ...   \n",
       "NBC_01798        custom      NaN   NaN     NaN  NBC_01798   \n",
       "NBC_01799        custom      NaN   NaN     NaN  NBC_01799   \n",
       "NBC_01800        custom      NaN   NaN     NaN  NBC_01800   \n",
       "NBC_01803        custom      NaN   NaN     NaN  NBC_01803   \n",
       "NBC_01808        custom      NaN   NaN     NaN  NBC_01808   \n",
       "\n",
       "                closest_placement_reference  \n",
       "genome_id                                    \n",
       "GCF_000009765.2                         NaN  \n",
       "GCF_000010605.1                         NaN  \n",
       "GCF_000091305.1                         NaN  \n",
       "GCF_000092385.1                         NaN  \n",
       "GCF_000147815.2                         NaN  \n",
       "...                                     ...  \n",
       "NBC_01798                               NaN  \n",
       "NBC_01799                               NaN  \n",
       "NBC_01800                               NaN  \n",
       "NBC_01803                               NaN  \n",
       "NBC_01808                               NaN  \n",
       "\n",
       "[2371 rows x 6 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_samples_mq_ncbi = df_samples_strepto_ncbi_all.loc[df_strepto_mq[df_strepto_mq.source == \"NCBI\"].index,:] \n",
    "df_samples_mq_nbc = df_samples_NBC_902.loc[df_strepto_mq[df_strepto_mq.source == \"NBC\"].index,:] \n",
    "df_samples_mq_strepo = pd.concat([df_samples_mq_ncbi, df_samples_mq_nbc])\n",
    "df_samples_mq_strepo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "a69638a9-a92f-4c4d-ae49-5a5a3899cb6b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "species\n",
       "sp.                   415\n",
       "albidoflavus          109\n",
       "anulatus               58\n",
       "olivaceus              46\n",
       "bacillaris             33\n",
       "canus                  29\n",
       "sp003846175            26\n",
       "californicus           24\n",
       "anthocyanicus          24\n",
       "papulosus              24\n",
       "rochei                 21\n",
       "mirabilis              21\n",
       "parvus                 20\n",
       "cellulosae             18\n",
       "clavuligerus           18\n",
       "microflavus            18\n",
       "virginiae_A            18\n",
       "libani                 15\n",
       "scabiei                15\n",
       "phaeochromogenes       15\n",
       "cyaneofuscatus         14\n",
       "silvae                 14\n",
       "griseoincarnatus       14\n",
       "griseoaurantiacus      13\n",
       "platensis              13\n",
       "fimicarius             12\n",
       "chartreusis            12\n",
       "griseus                12\n",
       "goshikiensis           12\n",
       "globisporus            12\n",
       "sp900116325            11\n",
       "sp001905735            11\n",
       "rubiginosohelvolus     11\n",
       "fungicidicus           11\n",
       "longwoodensis          11\n",
       "sp900105515            10\n",
       "sp000772045            10\n",
       "diastaticus            10\n",
       "halstedii              10\n",
       "sp000177195             9\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_strepto_mq.species.value_counts()[:40]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "8dbc2ccd-ee4d-41c7-9678-a1476f91466f",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'species_counts' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[33], line 6\u001b[0m\n\u001b[1;32m      3\u001b[0m df \u001b[38;5;241m=\u001b[39m df_strepto_mq\u001b[38;5;241m.\u001b[39mcopy()\n\u001b[1;32m      5\u001b[0m \u001b[38;5;66;03m# filter top 20 species\u001b[39;00m\n\u001b[0;32m----> 6\u001b[0m top_species_names \u001b[38;5;241m=\u001b[39m \u001b[43mspecies_counts\u001b[49m[:\u001b[38;5;241m20\u001b[39m]\u001b[38;5;241m.\u001b[39mindex\n\u001b[1;32m      7\u001b[0m df_top_species \u001b[38;5;241m=\u001b[39m df[df[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mspecies\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m.\u001b[39misin(top_species_names)]\n\u001b[1;32m      9\u001b[0m \u001b[38;5;66;03m# create a new column combining 'source' and 'quality'\u001b[39;00m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'species_counts' is not defined"
     ]
    }
   ],
   "source": [
    "import plotly.express as px\n",
    "\n",
    "df = df_strepto_mq.copy()\n",
    "\n",
    "# filter top 20 species\n",
    "top_species_names = species_counts[:20].index\n",
    "df_top_species = df[df['species'].isin(top_species_names)]\n",
    "\n",
    "# create a new column combining 'source' and 'quality'\n",
    "df_top_species['source_quality'] = df_top_species['source'] + '-' + df_top_species['quality']\n",
    "\n",
    "# compute total counts for each species\n",
    "species_total_counts = df_top_species['species'].value_counts()\n",
    "\n",
    "# sort species by total counts\n",
    "df_top_species['species'] = pd.Categorical(df_top_species['species'], categories=species_total_counts.index, ordered=True)\n",
    "\n",
    "fig = px.histogram(df_top_species, x=\"species\", color=\"source_quality\",\n",
    "                   barmode=\"stack\",\n",
    "                   title=\"Distribution of Top 20 Species by Source and Quality\",\n",
    "                   labels={'species':'Species', 'source_quality':'Source and Quality', 'count':'Count'})\n",
    "\n",
    "fig.update_layout(height=600, showlegend=True)\n",
    "fig.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "8b9f3d54-d0c6-4c31-8620-03b3c00bd4ca",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Write samples.csv table to config directory of qc_strepto_ncbi project\n",
    "project_name_4 = \"mq_strepto\"\n",
    "config_dir_4 = bgcflow_dir / f\"config/{project_name_4}\"\n",
    "\n",
    "samples_mq_strepto_table = config_dir_4 / \"samples.csv\"\n",
    "# df_samples_mq_strepo.to_csv(samples_table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "302ebf5d-c6c6-40bd-82b6-ac217e5ddc5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "processed_dir_4 = bgcflow_dir / \"data\" / \"processed\" / project_name_4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "0c65ff33-88db-42c0-bc72-e7c0e8968bc5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# df_stats.to_csv(processed_dir_1 / \"tables\" / \"df_filters.csv\")\n",
    "# df_stats.to_csv(processed_dir_2 / \"tables\" / \"df_filters.csv\")\n",
    "# df_strepto_mq.to_csv(processed_dir_4 / \"tables\" / \"df_filters.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3e111feb-3173-42eb-b9bf-7a8e61ed5f3d",
   "metadata": {},
   "source": [
    "# 5. Streptomyces phylogroup projects \n",
    "\n",
    "We used the MASH based clustering to define consistent phylogroups. Here a separate project is created for each of the phylgroups for detailed analysis. "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e6c6c46c-092c-407c-b516-2e2e363a247d",
   "metadata": {},
   "source": [
    "### Setup the config project using BGCFlow wrapper\n",
    "\n",
    "Follow the guide to set up a project called \"mq_strepto\" with samples tables and project config with the following rules\n",
    "\n",
    "```\n",
    "cd bgcflow_dir\n",
    "bgcflow init --project P1\n",
    "cp .examples/_genome_project_example/prokka-db.csv config/P1/\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "de3e46b5-5a47-4e04-947c-a62462b59c14",
   "metadata": {},
   "source": [
    "## Update project_config\n",
    "\n",
    "Update the file `config/P1/project_config.yaml` to allow seqfu rule on.\n",
    "\n",
    "```\n",
    "name: P1\n",
    "pep_version: 2.1.0\n",
    "description: 'Project to of individual phylogroups defined using MASH.'\n",
    "sample_table: samples.csv\n",
    "prokka-db: 'prokka-db.csv'\n",
    "rules:\n",
    "  seqfu: true\n",
    "  bigscape: true\n",
    "\n",
    "#### CUSTOM RULE CONFIGURATION ####\n",
    "rule_p  install_gtdbtk:\n",
    "    release: \"214\"\n",
    "    release_version: \"214\"\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "58a81928-d72b-49a4-b2fb-7815f53ef454",
   "metadata": {},
   "outputs": [],
   "source": [
    "samples_mq = bgcflow_dir / \"config\" / \"mq_strepto\" / \"samples.csv\"\n",
    "df_samples_mq = pd.read_csv(samples_mq, index_col= 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "0881759a-57fa-4524-a7a7-1c4d53795973",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_mash_clusters = pd.read_csv(\"assets/tables/df_mash_clusters_main_reduced.csv\", index_col=0)\n",
    "cluster_list = sorted(df_mash_clusters.Cluster.unique())\n",
    "for cluster in cluster_list:\n",
    "    phylogroup = \"P\" + str(cluster)\n",
    "    genome_list = df_mash_clusters[df_mash_clusters.Cluster == cluster].index.tolist()\n",
    "    df_samples_phylogroup = df_samples_mq.loc[genome_list, :]\n",
    "    samples_phylo_table = bgcflow_dir / \"config\" / phylogroup / \"samples.csv\"\n",
    "    df_samples_phylogroup.to_csv(samples_phylo_table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "034c044e-e87f-4571-9036-a1cf53d48275",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Write samples.csv table to config directory of qc_strepto_ncbi project\n",
    "project_name_4 = \"mq_strepto\"\n",
    "config_dir_4 = bgcflow_dir / f\"config/{project_name_4}\"\n",
    "\n",
    "samples_mq_strepto_table = config_dir_4 / \"samples.csv\"\n",
    "# df_samples_mq_strepo.to_csv(samples_table)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
